# NIEHS_NGS_Workshop (Currently Under Construction)<br>
This is 2-day (4hr each day) workshop on NGS data analysis.<br> 
We will use biowulf HPC as the computation evironment, and use jupyter lab as a interface  <br>

**This repository is for illustration only. Data files are included in biowulf folders for use in the class.**<br>

Target audience: NIEHS researchers who want to learn NGS data analysis.

Aims: to provide an overview of NGS analysis workflow for scRNA-seq, bulk RNA-seq and ChIP-seq. <br>
Prerequisite: confortable with shell scripts, R and python codes.  

Agenda of workshop, 

The README.md contains information to set up the workplace. <br>

Briefly,

On the 1st day, I will start from NGS_Workshop_Day1.pdf from the resources folder, do a ~30mins presentation.<br>
This will help setting up the working environment and get background knowledge on single-cell RNA-seq. <br>
The presentation will be followed by practical exercises in the scRNA-seq/notebooks folder.<br>
 
On the 2nd day, I will start from the NGS_Workshop_Day2.pdf file to do a presentation, which will be followed by exercise in the bulk-RNA and ChIP-seq folder. 


## This training will be divided into 3 parts. 

Bulk-RNA-seq <br>
Chip-seq<br>
Single cell RNA-seq <br> 

Each part will be given with a 15-30 minutes background introduction. 
then with hands-on practice/demo. 

To start with jupyter lab.

# We will use jupyter lab interface on Biowulf. (NIH) Please refer to "getting_started.md" to setup the working environment.

# The detailed plan of this workshop
<br>
<img src="resources/day1schedule.png">
<br>
<img src="resources/day2schedule.png">
