{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "8223958b-7418-484d-b692-7fe3834a5abc",
   "metadata": {},
   "source": [
    " ChIP-seq analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c72d39cd-87c6-4637-9b03-af59a26e6062",
   "metadata": {},
   "source": [
    "# prepare of input files"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fafedf01-7e07-436c-abea-9b7c344cf73d",
   "metadata": {},
   "source": [
    "## differenital peaks prepared previously"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "ce0b1a23-b32d-46e7-8f77-afd2c5f1f771",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# HOMER Differential Peaks\n",
      "# Original Peak File = peak/STAT1_30m_IFNa_peaks.narrowPeak\n",
      "# Peak File Tag Directory = STAT1_30m_IFNa_tagdir/\n",
      "# Background Tag Directory = STAT1_6h_IFNa_tagdir/\n",
      "#\n",
      "# Mode = Peaks enriched over background\n",
      "# Differential peaks = 25\n",
      "# Fraction of original peaks = 3.36%\n",
      "# Original number of peaks = 745\n",
      "# Size of region used to count tags = given peak sizes (-fixed)\n",
      "# Fold change cutoff = 4.00\n",
      "# Poisson p-value cutoff = 1.00e-04\n",
      "# strand = 2 (tags counted on 0=+strand, 1=-strand, 2=both strands)\n",
      "# peak directory tagAdjust = 51\n",
      "# background directory tagAdjust = 62\n",
      "#\n",
      "# cmd = getDifferentialPeaks peak/STAT1_30m_IFNa_peaks.narrowPeak STAT1_30m_IFNa_tagdir/ STAT1_6h_IFNa_tagdir/\n",
      "#\n",
      "#PeakID\tchr\tstart\tend\tstrand\tscore\tfocus ratio/other\tTotal Tags\tBackground Tags\tFold Change vs. Background\tp-value\n",
      "STAT1_30m_IFNa_peak_109\tchr2\t26980872\t26980945\t+\t0.000\t0.000\t16.0\t1.0\t16.23\t1.42e-14\n",
      "STAT1_30m_IFNa_peak_160\tchr3\t196569711\t196569757\t+\t0.000\t0.000\t31.0\t5.9\t5.24\t3.44e-13\n",
      "STAT1_30m_IFNa_peak_174\tchr5\t43040445\t43040515\t+\t0.000\t0.000\t48.0\t10.8\t4.43\t9.91e-17\n",
      "STAT1_30m_IFNa_peak_178\tchr5\t172483282\t172483308\t+\t0.000\t0.000\t21.0\t3.9\t5.32\t1.51e-09\n",
      "STAT1_30m_IFNa_peak_179\tchr5\t172483370\t172483413\t+\t0.000\t0.000\t28.0\t6.9\t4.06\t1.34e-09\n",
      "STAT1_30m_IFNa_peak_195\tchr7\t4660229\t4660263\t+\t0.000\t0.000\t31.0\t5.9\t5.24\t3.44e-13\n"
     ]
    }
   ],
   "source": [
    "cat IFNa30m_diff_peaks.txt | head -n 25\n",
    "# could be open in R for better look."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c28ad873-e021-4622-8d29-b187ffca74e6",
   "metadata": {},
   "source": [
    "## file 2, refseq annotation files to tell the gene locations."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a5a397e5-51aa-497c-a809-aadf9381b268",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-06-15 08:14:45--  http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz\n",
      "Resolving dtn06-e0 (dtn06-e0)... 10.1.200.242\n",
      "Connecting to dtn06-e0 (dtn06-e0)|10.1.200.242|:3128... connected.\n",
      "Proxy request sent, awaiting response... 200 OK\n",
      "Length: 8043766 (7.7M) [application/x-gzip]\n",
      "Saving to: ‘hg19.refGene.txt.gz’\n",
      "\n",
      "100%[======================================>] 8,043,766   10.2MB/s   in 0.8s   \n",
      "\n",
      "2021-06-15 08:14:46 (10.2 MB/s) - ‘hg19.refGene.txt.gz’ saved [8043766/8043766]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Downloaded hg19.refgene.txt : https://www.biostars.org/p/156637/\n",
    "wget -c -O hg19.refGene.txt.gz http://hgdownload.soe.ucsc.edu/goldenPath/hg19/database/refGene.txt.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "53b1d33f-80cc-432b-b97a-3e026604844a",
   "metadata": {},
   "source": [
    "# use CEAS to add gene annotation to the peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "903b22e0-90b8-4d5b-925e-69b86320b905",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[+] Loading ceas  1.0.2  on cn2953 \n"
     ]
    }
   ],
   "source": [
    "module load ceas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0196d245-e60c-4eb1-bcf3-eceba9eca55b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Tue, 15 Jun 2021 08:18:56: \n",
      "# ARGUMENTS: \n",
      "# name = STAT1_6h_IFNa_bed_CEAS\n",
      "# gene annotation table = /fdb/CEAS/hg19.refGene\n",
      "# BED file = STAT1_6h_IFNa_peaks.bed\n",
      "# WIG file = None\n",
      "# extra BED file = None\n",
      "# ChIP annotation = On\n",
      "# gene-centered annotation =  On\n",
      "# average profiling = Off\n",
      "# dump profiles = Off\n",
      "# re-annotation for genome background (ChIP region annotation) = False\n",
      "# promoter sizes (ChIP region annotation) = 1000,2000,3000 bp\n",
      "# downstream sizes (ChIP region annotation) = 1000,2000,3000 bp\n",
      "# bidrectional promoter sizes (ChIP region annotation) = 2500,5000 bp\n",
      "# span size (gene-centered annotation) = 3000 bp \n",
      "INFO  @ Tue, 15 Jun 2021 08:18:56: #1 read the gene table... \n",
      "INFO  @ Tue, 15 Jun 2021 08:18:59: #2 read the bed file of ChIP regions... \n",
      "INFO  @ Tue, 15 Jun 2021 08:18:59: #3 perform gene-centered annotation... \n",
      "INFO  @ Tue, 15 Jun 2021 08:19:00: #4 See STAT1_6h_IFNa_bed_CEAS.xls for gene-centered annotation! \n",
      "INFO  @ Tue, 15 Jun 2021 08:19:00: #5 read the pre-computed genome bg annotation... \n",
      "INFO  @ Tue, 15 Jun 2021 08:19:00: #6 perform ChIP region annotation... \n",
      "INFO  @ Tue, 15 Jun 2021 08:19:00: #7 write a R script of ChIP region annotation... \n",
      "\n",
      "R version 4.0.5 (2021-03-31) -- \"Shake and Throw\"\n",
      "Copyright (C) 2021 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # ARGUMENTS: \n",
      "> # name = STAT1_6h_IFNa_bed_CEAS\n",
      "> # gene annotation table = /fdb/CEAS/hg19.refGene\n",
      "> # BED file = STAT1_6h_IFNa_peaks.bed\n",
      "> # WIG file = None\n",
      "> # extra BED file = None\n",
      "> # ChIP annotation = On\n",
      "> # gene-centered annotation =  On\n",
      "> # average profiling = Off\n",
      "> # dump profiles = Off\n",
      "> # re-annotation for genome background (ChIP region annotation) = False\n",
      "> # promoter sizes (ChIP region annotation) = 1000,2000,3000 bp\n",
      "> # downstream sizes (ChIP region annotation) = 1000,2000,3000 bp\n",
      "> # bidrectional promoter sizes (ChIP region annotation) = 2500,5000 bp\n",
      "> # span size (gene-centered annotation) = 3000 bp\n",
      "> pdf(\"STAT1_6h_IFNa_bed_CEAS.pdf\",height=11.5,width=8.5)\n",
      "> \n",
      "> # 08:19:00 Tue, 15 Jun 2021\n",
      "> # \n",
      "> # ChIP annotation\n",
      "> # \n",
      "> \n",
      "> \n",
      "> # \n",
      "> # Chromosomal Distribution\n",
      "> # \n",
      "> \n",
      "> par(mar=c(4, 4, 5, 3.8),oma=c(4, 2, 4, 2))\n",
      "> r0<-c(0.5772044730953612, 4.22972183331696, 0.001018581691264085, 1.2267343823047572, 1.2419026590565934, 2.130934258467313, 1.6312524425251476, 2.7930307658918228, 2.7901529658122874, 2.7553677874521907, 2.9271522032873065, 3.09901025153384, 3.5415410441480915, 4.552906759093342, 4.6296133236866694, 4.740098757017399, 4.071019442576874, 5.083587820241931, 5.443171701395292, 6.041293915367275, 6.320600059973599, 6.592426378786846, 6.949844239837699, 8.692686601864827, 7.9377273515753135)\n",
      "> r1<-c(36.68202764976958, 1.6589861751152075, 0.0, 2.1198156682027647, 2.3963133640553, 1.2903225806451613, 1.566820276497696, 0.7373271889400922, 3.686635944700461, 2.1198156682027647, 1.935483870967742, 1.4746543778801844, 0.9216589861751152, 3.410138248847926, 3.5944700460829497, 3.963133640552995, 2.3041474654377883, 1.6589861751152075, 3.5944700460829497, 4.7926267281106, 2.3963133640553, 3.963133640552995, 2.672811059907834, 5.8986175115207375, 5.161290322580645)\n",
      "> height<-rbind(r0,r1)\n",
      "> names=c(\"Y\",\"X\",\"M\",\"22\",\"21\",\"20\",\"19\",\"18\",\"17\",\"16\",\"15\",\"14\",\"13\",\"12\",\"11\",\"10\",\"9\",\"8\",\"7\",\"6\",\"5\",\"4\",\"3\",\"2\",\"1\")\n",
      "> mp<-barplot(height=height,names=names,beside=TRUE,horiz=TRUE,col=c(\"#5FA1C1\",\"#EB9D86\"),main=\"Chromosomal Distribution of ChIP Regions\",xlab=\"Percentage %\",ylab=\"Chromosome\",border=FALSE,xlim=c(0.000000,67.250384),cex.names=0.8)\n",
      "> text(x=c(0.5772044730953612, 4.22972183331696, 0.001018581691264085, 1.2267343823047572, 1.2419026590565934, 2.130934258467313, 1.6312524425251476, 2.7930307658918228, 2.7901529658122874, 2.7553677874521907, 2.9271522032873065, 3.09901025153384, 3.5415410441480915, 4.552906759093342, 4.6296133236866694, 4.740098757017399, 4.071019442576874, 5.083587820241931, 5.443171701395292, 6.041293915367275, 6.320600059973599, 6.592426378786846, 6.949844239837699, 8.692686601864827, 7.9377273515753135),y=mp[1,],label=c(\"0.6 %\",\"4.2 %\",\"0.001 %\",\"1.2 %\",\"1.2 %\",\"2.1 %\",\"1.6 %\",\"2.8 %\",\"2.8 %\",\"2.8 %\",\"2.9 %\",\"3.1 %\",\"3.5 %\",\"4.6 %\",\"4.6 %\",\"4.7 %\",\"4.1 %\",\"5.1 %\",\"5.4 %\",\"6.0 %\",\"6.3 %\",\"6.6 %\",\"6.9 %\",\"8.7 %\",\"7.9 %\"),pos=4,offset=0.2,cex=0.7)\n",
      "> text(x=c(36.68202764976958, 1.6589861751152075, 0.0, 2.1198156682027647, 2.3963133640553, 1.2903225806451613, 1.566820276497696, 0.7373271889400922, 3.686635944700461, 2.1198156682027647, 1.935483870967742, 1.4746543778801844, 0.9216589861751152, 3.410138248847926, 3.5944700460829497, 3.963133640552995, 2.3041474654377883, 1.6589861751152075, 3.5944700460829497, 4.7926267281106, 2.3963133640553, 3.963133640552995, 2.672811059907834, 5.8986175115207375, 5.161290322580645),y=mp[2,],label=c(\"36.7 % (4.3e-322)\",\"1.7 % (1.6e-06)\",\"0.000 % (0.989)\",\"2.1 % (0.005)\",\"2.4 % (7.0e-04)\",\"1.3 % (0.028)\",\"1.6 % (0.496)\",\"0.7 % (1.3e-06)\",\"3.7 % (0.034)\",\"2.1 % (0.115)\",\"1.9 % (0.027)\",\"1.5 % (5.0e-04)\",\"0.9 % (3.6e-08)\",\"3.4 % (0.037)\",\"3.6 % (0.056)\",\"4.0 % (0.127)\",\"2.3 % (0.001)\",\"1.7 % (2.9e-09)\",\"3.6 % (0.003)\",\"4.8 % (0.045)\",\"2.4 % (1.5e-09)\",\"4.0 % (1.2e-04)\",\"2.7 % (3.1e-10)\",\"5.9 % (3.7e-04)\",\"5.2 % (2.2e-04)\"),pos=4,offset=0.2,cex=0.7)\n",
      "> legend(\"right\",legend=c(\"Genome\",\"ChIP (p-value)\"),col=c(\"#5FA1C1\",\"#EB9D86\"),pch=15,bty=\"n\")\n",
      "> \n",
      "> # \n",
      "> # Promoter,Bipromoter,Downstream, Gene and Regions of interest\n",
      "> # \n",
      "> \n",
      "> par(mfrow=c(4, 1),mar=c(4, 4, 5, 3.8),oma=c(4, 2, 4, 2))\n",
      "> r0<-c(1.0778189662488047, 1.8458418335304945, 2.5041401357327597)\n",
      "> r1<-c(7.741935483870968, 8.940092165898617, 10.046082949308756)\n",
      "> height<-rbind(r0,r1)\n",
      "> names=c(\"<=1000 bp\",\"<=2000 bp\",\"<=3000 bp\")\n",
      "> mp<-barplot(height=height,names=names,beside=TRUE,horiz=FALSE,col=c(\"#5FA1C1\",\"#EB9D86\"),main=\"Promoter\",ylab=\"Percentage %\",border=FALSE,ylim=c(0.000000,18.417819),cex.names=1)\n",
      "> text(x=mp[1,],y=c(1.0778189662488047, 1.8458418335304945, 2.5041401357327597),label=c(\"1.1 %\",\"1.8 %\",\"2.5 %\"),pos=3,offset=0.2)\n",
      "> text(x=mp[2,],y=c(7.741935483870968, 8.940092165898617, 10.046082949308756),label=c(\"7.7 %\n",
      "+ (1.6e-44)\",\"8.9 %\n",
      "+ (5.3e-37)\",\"10.0 %\n",
      "+ (2.7e-34)\"),pos=3,offset=0.2)\n",
      "> legend(\"topleft\",legend=c(\"Genome\",\"ChIP (p-value)\"),col=c(\"#5FA1C1\",\"#EB9D86\"),pch=15,bty=\"n\")\n",
      "> r0<-c(0.03315299324036055, 0.07115345356565259)\n",
      "> r1<-c(0.09216589861751152, 0.18433179723502305)\n",
      "> height<-rbind(r0,r1)\n",
      "> names=c(\"<=2500 bp\",\"<=5000 bp\")\n",
      "> mp<-barplot(height=height,names=names,beside=TRUE,horiz=FALSE,col=c(\"#5FA1C1\",\"#EB9D86\"),main=\"Bidirectional Promoter\",ylab=\"Percentage %\",border=FALSE,ylim=c(0.000000,0.337942),cex.names=1)\n",
      "> text(x=mp[1,],y=c(0.03315299324036055, 0.07115345356565259),label=c(\"0.03 %\",\"0.07 %\"),pos=3,offset=0.2)\n",
      "> text(x=mp[2,],y=c(0.09216589861751152, 0.18433179723502305),label=c(\"0.09 %\n",
      "+ (0.051)\",\"0.2 %\n",
      "+ (0.043)\"),pos=3,offset=0.2)\n",
      "> legend(\"topleft\",legend=c(\"Genome\",\"ChIP (p-value)\"),col=c(\"#5FA1C1\",\"#EB9D86\"),pch=15,bty=\"n\")\n",
      "> r0<-c(0.9730461808371527, 1.745775386415225, 2.4130875229817494)\n",
      "> r1<-c(0.9216589861751152, 1.4746543778801844, 2.1198156682027647)\n",
      "> height<-rbind(r0,r1)\n",
      "> names=c(\"<=1000 bp\",\"<=2000 bp\",\"<=3000 bp\")\n",
      "> mp<-barplot(height=height,names=names,beside=TRUE,horiz=FALSE,col=c(\"#5FA1C1\",\"#EB9D86\"),main=\"Downstream\",ylab=\"Percentage %\",border=FALSE,ylim=c(0.000000,4.423994),cex.names=1)\n",
      "> text(x=mp[1,],y=c(0.9730461808371527, 1.745775386415225, 2.4130875229817494),label=c(\"1.0 %\",\"1.7 %\",\"2.4 %\"),pos=3,offset=0.2)\n",
      "> text(x=mp[2,],y=c(0.9216589861751152, 1.4746543778801844, 2.1198156682027647),label=c(\"0.9 %\n",
      "+ (0.513)\",\"1.5 %\n",
      "+ (0.295)\",\"2.1 %\n",
      "+ (0.306)\"),pos=3,offset=0.2)\n",
      "> legend(\"topleft\",legend=c(\"Genome\",\"ChIP (p-value)\"),col=c(\"#5FA1C1\",\"#EB9D86\"),pch=15,bty=\"n\")\n",
      "> r0<-c(0.4375912850150495, 1.477526375589972, 1.9330962410592611, 42.376360903373964, 46.22457480503825)\n",
      "> r1<-c(5.345622119815668, 0.5529953917050692, 0.8294930875576038, 20.737327188940093, 27.46543778801843)\n",
      "> height<-rbind(r0,r1)\n",
      "> names=c(\"5'UTR\",\"3'UTR\",\"Coding Exon\",\"Intron\",\"All\")\n",
      "> mp<-barplot(height=height,names=names,beside=TRUE,horiz=FALSE,col=c(\"#5FA1C1\",\"#EB9D86\"),main=\"Gene\",ylab=\"Percentage %\",border=FALSE,ylim=c(0.000000,84.745054),cex.names=1)\n",
      "> text(x=mp[1,],y=c(0.4375912850150495, 1.477526375589972, 1.9330962410592611, 42.376360903373964, 46.22457480503825),label=c(\"0.4 %\",\"1.5 %\",\"1.9 %\",\"42.4 %\",\"46.2 %\"),pos=3,offset=0.2)\n",
      "> text(x=mp[2,],y=c(5.345622119815668, 0.5529953917050692, 0.8294930875576038, 20.737327188940093, 27.46543778801843),label=c(\"5.3 %\n",
      "+ (1.4e-43)\",\"0.6 %\n",
      "+ (0.004)\",\"0.8 %\n",
      "+ (0.003)\",\"20.7 %\n",
      "+ (2.6e-51)\",\"27.5 %\n",
      "+ (6.0e-37)\"),pos=3,offset=0.2)\n",
      "> legend(\"topleft\",legend=c(\"Genome\",\"ChIP (p-value)\"),col=c(\"#5FA1C1\",\"#EB9D86\"),pch=15,bty=\"n\")\n",
      "> \n",
      "> # \n",
      "> # Distribution of Genome and ChIP regions over cis-regulatory element\n",
      "> # Note that the x may be modified for better graphics in case a value is too small\n",
      "> # Thus, look at the labels of the pie chart to get the real percentage values\n",
      "> # \n",
      "> \n",
      "> par(mfcol=c(2, 2),mar=c(3, 3, 4, 2.8),oma=c(4, 2, 4, 2))\n",
      "> x<-c(0.010778,0.010000,0.010000,0.010000,0.010000,0.010000,0.010000,0.014097,0.018732,0.415634,0.500274)\n",
      "> pie(x=x,labels=c(\"1.1 %\",\"0.7 %\",\"0.6 %\",\"0.9 %\",\"0.7 %\",\"0.6 %\",\"0.4 %\",\"1.4 %\",\"1.9 %\",\"41.6 %\",\"50.0 %\"),main=\"Genome\",col=c(\"#445FA2\",\"#EB9D86\",\"#799F7A\",\"#6C527F\",\"#5FA1C1\",\"#E8BB77\",\"#A8C5EF\",\"#FDCDB9\",\"#C6E6B5\",\"#F1D5EE\",\"#B4E1F6\"),clockwise=TRUE,border=FALSE,radius=0.9,cex=0.8,init.angle=90,density=100)\n",
      "> x<-c(0.000000,1.000000)\n",
      "> y<-c(0.000000,1.000000)\n",
      "> plot(x, y,type=\"n\",main=\"\",xlab=\"\",ylab=\"\",frame=FALSE,axes=FALSE,xaxt=\"s\",yaxt=\"s\")\n",
      "> legend(\"top\",legend=c(\"Promoter (<=1000 bp): 1.1 %\",\"Promoter (1000-2000 bp): 0.7 %\",\"Promoter (2000-3000 bp): 0.6 %\",\"Downstream (<=1000 bp): 0.9 %\",\"Downstream (1000-2000 bp): 0.7 %\",\"Downstream (2000-3000 bp): 0.6 %\",\"5'UTR: 0.4 %\",\"3'UTR: 1.4 %\",\"Coding exon: 1.9 %\",\"Intron: 41.6 %\",\"Distal intergenic: 50.0 %\"),col=c(\"#445FA2\",\"#EB9D86\",\"#799F7A\",\"#6C527F\",\"#5FA1C1\",\"#E8BB77\",\"#A8C5EF\",\"#FDCDB9\",\"#C6E6B5\",\"#F1D5EE\",\"#B4E1F6\"),pch=15,bty=\"n\")\n",
      "> x<-c(0.077419,0.011982,0.011060,0.010000,0.010000,0.010000,0.047005,0.010000,0.010000,0.193548,0.629493)\n",
      "> pie(x=x,labels=c(\"7.7 %\",\"1.2 %\",\"1.1 %\",\"0.6 %\",\"0.3 %\",\"0.6 %\",\"4.7 %\",\"0.6 %\",\"0.8 %\",\"19.4 %\",\"62.9 %\"),main=\"ChIP\",col=c(\"#445FA2\",\"#EB9D86\",\"#799F7A\",\"#6C527F\",\"#5FA1C1\",\"#E8BB77\",\"#A8C5EF\",\"#FDCDB9\",\"#C6E6B5\",\"#F1D5EE\",\"#B4E1F6\"),clockwise=TRUE,border=FALSE,radius=0.9,cex=0.8,init.angle=90,density=100)\n",
      "> x<-c(0.000000,1.000000)\n",
      "> y<-c(0.000000,1.000000)\n",
      "> plot(x, y,type=\"n\",main=\"\",xlab=\"\",ylab=\"\",frame=FALSE,axes=FALSE,xaxt=\"s\",yaxt=\"s\")\n",
      "> legend(\"top\",legend=c(\"Promoter (<=1000 bp): 7.7 %\",\"Promoter (1000-2000 bp): 1.2 %\",\"Promoter (2000-3000 bp): 1.1 %\",\"Downstream (<=1000 bp): 0.6 %\",\"Downstream (1000-2000 bp): 0.3 %\",\"Downstream (2000-3000 bp): 0.6 %\",\"5'UTR: 4.7 %\",\"3'UTR: 0.6 %\",\"Coding exon: 0.8 %\",\"Intron: 19.4 %\",\"Distal intergenic: 62.9 %\"),col=c(\"#445FA2\",\"#EB9D86\",\"#799F7A\",\"#6C527F\",\"#5FA1C1\",\"#E8BB77\",\"#A8C5EF\",\"#FDCDB9\",\"#C6E6B5\",\"#F1D5EE\",\"#B4E1F6\"),pch=15,bty=\"n\")\n",
      "> \n",
      "> # \n",
      "> # ChIP regions over the genome\n",
      "> # \n",
      "> \n",
      "> par(mar=c(4, 4, 5, 3.8),oma=c(4, 2, 4, 2))\n",
      "> layout(matrix(c(1, 0, 2, 2), 2, 2, byrow = TRUE),widths=c(1, 1),heights=c(1, 5))\n",
      "> x<-c(0.000000,1.000000)\n",
      "> y<-c(0.000000,1.000000)\n",
      "> plot(x, y,type=\"n\",main=\"Distribution of Peak Heights\",xlab=\"\",ylab=\"\",xlim=c(0.000000,1.000000),ylim=c(0.000000,1.000000),frame=FALSE,xaxt=\"s\",yaxt=\"n\",cex=0.9)\n",
      "> x<-c(0.000000,1.000000,1.000000,0.000000)\n",
      "> y<-c(0.000000,0.000000,1.000000,1.000000)\n",
      "> polygon(x,y,col=c(\"black\"))\n",
      "> x <- c(0.000000,0.075000,0.150000,0.225000,0.300000,0.375000,0.450000,0.525000,0.600000,0.675000,0.750000,0.825000,0.900000,0.975000)\n",
      "> y<-c(0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.800000)\n",
      "> lines(x, y,xlim=c(0, 1.0),ylim=c(0, 1),type=\"l\",col=c(\"cyan\"),lwd=2)\n",
      "> x<-c(307991.000000,247523670.000000)\n",
      "> y<-c(0.600000,24.400000)\n",
      "> plot(x, y,type=\"n\",main=\"ChIP Regions (Peaks) over Chromosomes\",xlab=\"Chromosome Size (bp)\",ylab=\"Chromosome\",xlim=c(307991.000000,247523670.000000),ylim=c(0.600000,24.400000),frame=FALSE,xaxt=\"s\",yaxt=\"n\")\n",
      "> start <- c(948426, 1710340, 4363309, 6812449, 8812619, 11791916, 21669948, 24285527, 25423554, 27998607, 29216966, 31714236, 36555360, 41311003, 48036490, 52868134, 54842838, 71976206, 79085928, 89449668, 93166811, 99839769, 100531097, 109568543, 113480959, 120759200, 143992742, 145438232, 149934141, 150539988, 154580560, 158979582, 160857704, 169337664, 173640049, 185303426, 196527635, 203490767, 205216596, 206990207, 212665066, 213243844, 234536657, 237077265, 247381547)\n",
      "> end <- c(948930, 1710462, 4363338, 6812547, 8812884, 11791943, 21670253, 24285593, 25423581, 27999035, 29217013, 31714408, 36555431, 41311050, 48036538, 52868209, 54843085, 71976280, 79086159, 89449704, 93166861, 99839895, 100531174, 109568593, 113481045, 120759330, 143992923, 145438539, 149934208, 150540025, 154580804, 158979659, 160857731, 169337837, 173640509, 185303586, 196527821, 203490794, 205216662, 206990402, 212665093, 213243889, 234536708, 237077335, 247381777)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 23.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(23.6, 23.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"1\",side=2,line=0,outer=FALSE,at=24.0)\n",
      "> start <- c(7005650, 9904215, 11640935, 12417652, 26558916, 30850204, 35223019, 37384088, 38879877, 45795126, 55920881, 58135961, 68964972, 75538008, 86221273, 86811416, 87465001, 89860433, 90376631, 97151842, 99176994, 109591201, 113299800, 113836180, 119542479, 120989900, 131914274, 136640358, 149372034, 152118236, 158678782, 163175136, 169856277, 179279420, 191878828, 201248499, 216696184, 227637324, 230786054, 231276601)\n",
      "> end <- c(7005677, 9904242, 11641034, 12417892, 26558994, 30850246, 35223046, 37384224, 38879906, 45795156, 55921264, 58136034, 68965018, 75538256, 86221369, 86811517, 87465055, 89860474, 90376661, 97152003, 99177124, 109591228, 113299997, 113836394, 119542545, 120990183, 131914330, 136640387, 149372093, 152118469, 158678809, 163175390, 169856364, 179279650, 191879052, 201248790, 216696412, 227637353, 230786115, 231276641)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 22.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(22.6, 22.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"2\",side=2,line=0,outer=FALSE,at=23.0)\n",
      "> start <- c(5182782, 28389876, 36949522, 38180127, 46152650, 48514313, 58311261, 69841144, 88083867, 90364931, 122282938, 128562227, 146262251, 151919097, 170902285, 171897600, 176825492, 183271502, 187086203, 197439264)\n",
      "> end <- c(5182862, 28389927, 36949765, 38180154, 46152966, 48514644, 58311490, 69841436, 88083937, 90364958, 122283295, 128562278, 146262658, 151919127, 170902324, 171897627, 176825519, 183271554, 187086273, 197439415)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 21.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(21.6, 21.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"3\",side=2,line=0,outer=FALSE,at=22.0)\n",
      "> start <- c(3049520, 17578542, 24586229, 26199189, 38285469, 49093266, 49634003, 54871759, 73919009, 76949508, 78656301, 81064712, 89300410, 110915743, 122722188, 140236158, 141758673, 144434672, 162007997, 169239788, 170548195, 185395913, 186153517, 190155063)\n",
      "> end <- c(3049550, 17578804, 24586264, 26199301, 38285659, 49093325, 49634030, 54871807, 73919036, 76949569, 78656334, 81064745, 89300768, 110915770, 122722403, 140236210, 141758718, 144434825, 162008026, 169240212, 170548283, 185396030, 186153544, 190155178)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 20.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(20.6, 20.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"4\",side=2,line=0,outer=FALSE,at=21.0)\n",
      "> start <- c(3323910, 5523163, 10627848, 39077074, 46218243, 49451669, 66692759, 77656188, 81581287, 84943579, 94879143, 99982503, 103424998, 125856038, 131746531, 134057059, 148293631, 162325027)\n",
      "> end <- c(3323937, 5523587, 10627917, 39077377, 46218270, 49451710, 66692799, 77656282, 81581345, 84944008, 94879242, 99982540, 103425044, 125856065, 131746647, 134057086, 148293662, 162325087)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 19.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(19.6, 19.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"5\",side=2,line=0,outer=FALSE,at=20.0)\n",
      "> start <- c(10838836, 16236904, 17910572, 24721585, 25963045, 28695523, 30457036, 32806484, 33546576, 34544490, 37400889, 42013415, 43597720, 47382451, 56934132, 67774352, 69434147, 80239146, 82462925, 84772526, 88707471, 99281920, 100705927, 109625747, 119788733, 126240287, 138043959, 143521825, 144053749, 147561778, 150881124, 163881537, 168241580, 169852248)\n",
      "> end <- c(10838966, 16237215, 17910679, 24721742, 25963095, 28695576, 30457197, 32806695, 33546621, 34544610, 37401144, 42013619, 43598016, 47382774, 56934527, 67774399, 69434174, 80239173, 82462999, 84772573, 88707510, 99282130, 100705970, 109625777, 119788760, 126240360, 138043998, 143521898, 144053860, 147561895, 150881266, 163881591, 168241836, 169852275)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 18.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(18.6, 18.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"6\",side=2,line=0,outer=FALSE,at=19.0)\n",
      "> start <- c(2445754, 8805216, 21540771, 33079296, 36444514, 38503326, 51172643, 56029811, 57555197, 61278123, 61779547, 73669322, 87834567, 92747241, 99942786, 100499676, 105924462, 113159687, 120640107, 128172632, 134832545, 138728965, 139762628, 143123626, 151047092)\n",
      "> end <- c(2445825, 8805340, 21540798, 33079380, 36444676, 38503478, 51172710, 56030080, 57555228, 61278151, 61779581, 73669371, 87834851, 92747591, 99942823, 100499757, 105924622, 113159993, 120640148, 128172659, 134832606, 138729181, 139762894, 143123786, 151047332)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 17.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(17.6, 17.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"7\",side=2,line=0,outer=FALSE,at=18.0)\n",
      "> start <- c(10487988, 20773297, 26150652, 30513624, 42074723, 43796123, 46947822, 68991669, 90996245, 99318498, 105630205, 124049894, 144105491, 145060505)\n",
      "> end <- c(10488206, 20773326, 26150679, 30513672, 42074846, 43796171, 46947849, 68991696, 90996283, 99318569, 105630277, 124050099, 144105822, 145060646)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 16.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(16.6, 16.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"8\",side=2,line=0,outer=FALSE,at=17.0)\n",
      "> start <- c(5450379, 14522678, 15296322, 26954225, 32526059, 33264981, 37197306, 66817058, 74510170, 90289626, 91381872, 100174410, 100881456, 104603111, 116648796, 123529137, 127129921, 135998604)\n",
      "> end <- c(5450406, 14522862, 15296397, 26954298, 32526473, 33265017, 37197382, 66817093, 74510201, 90289800, 91381912, 100174440, 100881603, 104603415, 116648831, 123529173, 127130026, 135998641)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 15.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(15.6, 15.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"9\",side=2,line=0,outer=FALSE,at=16.0)\n",
      "> start <- c(3871060, 6205773, 11646921, 25026036, 38787871, 42369022, 52086807, 53140504, 72338404, 80719077, 88535140, 91061738, 93971005, 98433719, 102027668, 103953103, 104541131, 105576322, 111827983, 112935630, 114851719)\n",
      "> end <- c(3871296, 6205855, 11647010, 25026278, 38787923, 42369058, 52086849, 53140540, 72338439, 80719139, 88535177, 91061796, 93971032, 98433754, 102027703, 103953190, 104541178, 105576385, 111828021, 112935657, 114851960)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 14.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(14.6, 14.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"10\",side=2,line=0,outer=FALSE,at=15.0)\n",
      "> start <- c(307991, 1347717, 4414695, 5645925, 30961186, 31634365, 34330915, 48803272, 57333549, 59278475, 60719849, 64764391, 65695659, 67771007, 77735595, 78356401, 80913290, 82503965, 102318035, 107940817, 113244706, 115417859, 117043807, 118895373)\n",
      "> end <- c(308056, 1347752, 4414792, 5645981, 30961216, 31634468, 34331124, 48803299, 57333963, 59278502, 60719878, 64764675, 65695826, 67771055, 77735732, 78356454, 80913330, 82504102, 102318073, 107940942, 113244844, 115417886, 117044124, 118895540)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 13.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(13.6, 13.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"11\",side=2,line=0,outer=FALSE,at=14.0)\n",
      "> start <- c(6445448, 7758763, 8477854, 10050826, 11802631, 13199115, 14437621, 31907484, 33741282, 34769653, 37993356, 47411059, 56754042, 62658627, 64612537, 72298469, 77198354, 82641893, 91956343, 96794995, 104201781, 113344583, 121476966, 125404857, 127567926, 129821981, 133199227)\n",
      "> end <- c(6445524, 7758800, 8477888, 10050864, 11802670, 13199142, 14437787, 31907515, 33741322, 34769681, 37993386, 47411088, 56754377, 62658705, 64612567, 72298500, 77198392, 82641926, 91956406, 96795062, 104201864, 113344888, 121477185, 125404922, 127568099, 129822157, 133199280)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 12.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(12.6, 12.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"12\",side=2,line=0,outer=FALSE,at=13.0)\n",
      "> start <- c(43566357, 48339775, 50069995, 67914464, 90946141, 112653842)\n",
      "> end <- c(43566584, 48339802, 50070050, 67914721, 90946193, 112653872)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 11.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(11.6, 11.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"13\",side=2,line=0,outer=FALSE,at=12.0)\n",
      "> start <- c(24630162, 33366552, 36830600, 58711909, 65548196, 68034062, 70168777, 89625238, 91580888, 94576774, 97284292, 100841721)\n",
      "> end <- c(24630606, 33366583, 36830629, 58711992, 65548306, 68034110, 70168804, 89625274, 91581037, 94577159, 97284482, 100841773)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 10.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(10.6, 10.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"14\",side=2,line=0,outer=FALSE,at=11.0)\n",
      "> start <- c(27641717, 34339844, 42263705, 43425648, 45020822, 50540162, 52024207, 67963097, 74287076, 75417717, 77905846, 81400259, 89183375, 90727450, 91590878, 98503588, 102193517)\n",
      "> end <- c(27641791, 34339928, 42263771, 43425755, 45021248, 50540221, 52024235, 67963152, 74287183, 75417905, 77905945, 81400480, 89183409, 90727602, 91590913, 98503848, 102193560)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 9.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(9.6, 9.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"15\",side=2,line=0,outer=FALSE,at=10.0)\n",
      "> start <- c(3313867, 9179222, 10595191, 11391385, 15400403, 17922033, 20186606, 22022320, 23568765, 31033927, 33865585, 35279665, 46433716, 48642502, 50699229, 64471782, 70438478)\n",
      "> end <- c(3313954, 9179249, 10595272, 11391662, 15400430, 17922094, 20186732, 22022471, 23568894, 31034089, 33865612, 35279695, 46433744, 48642684, 50699257, 64471809, 70438587)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 8.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(8.6, 8.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"16\",side=2,line=0,outer=FALSE,at=9.0)\n",
      "> start <- c(1992985, 6659268, 7482818, 14286786, 20610487, 22245426, 25958208, 26595312, 30454368, 31291037, 33571414, 38072828, 39647076, 40264541, 41158630, 45213954, 47816955, 54910023, 56444962, 60557448, 62015136, 76975958, 78237238, 80377472)\n",
      "> end <- c(1993030, 6659532, 7482845, 14286813, 20610554, 22245458, 25958263, 26595641, 30454699, 31291089, 33571659, 38072969, 39647130, 40264952, 41158862, 45213985, 47817165, 54910102, 56445000, 60557475, 62015163, 76976012, 78237457, 80377522)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 7.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(7.6, 7.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"17\",side=2,line=0,outer=FALSE,at=8.0)\n",
      "> start <- c(2641709, 10626593, 19230445, 21750477, 37880974, 46598993, 53144916, 60173511)\n",
      "> end <- c(2642067, 10626676, 19230472, 21750506, 37881004, 46599027, 53144960, 60173918)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 6.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(6.6, 6.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"18\",side=2,line=0,outer=FALSE,at=7.0)\n",
      "> start <- c(4841080, 9540788, 10196763, 11692719, 16254402, 17516404, 19772732, 40939735, 41726868, 47217748, 48018653, 49371898, 50676526, 51727695)\n",
      "> end <- c(4841130, 9540835, 10197061, 11692746, 16254451, 17516835, 19772930, 40939798, 41726920, 47217933, 48018942, 49371925, 50676587, 51727832)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 5.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(5.6, 5.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"19\",side=2,line=0,outer=FALSE,at=6.0)\n",
      "> start <- c(388893, 1306425, 29814111, 35579156, 36751336, 39676729, 46116141, 47894587, 48393019, 48983380, 58342862, 62205538)\n",
      "> end <- c(388920, 1306461, 29814162, 35579316, 36751614, 39676772, 46116176, 47894724, 48393156, 48983455, 58342924, 62205643)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 4.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(4.6, 4.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"20\",side=2,line=0,outer=FALSE,at=5.0)\n",
      "> start <- c(10800968, 15826627, 17104705, 22565658, 35029651, 36012667, 36855726, 38707630, 40159661, 42733741, 44261103, 44772782, 45275016, 46014069)\n",
      "> end <- c(10801019, 15826665, 17104960, 22565766, 35029754, 36012709, 36855774, 38707657, 40159786, 42733976, 44261130, 44772809, 45275071, 46014098)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 3.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(3.6, 3.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"21\",side=2,line=0,outer=FALSE,at=4.0)\n",
      "> start <- c(18632708, 19417359, 20284170, 23828972, 28329436, 35773024, 36635623, 39076577, 50977820)\n",
      "> end <- c(18632963, 19417580, 20284323, 23829007, 28329463, 35773054, 36635683, 39076613, 50978344)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 2.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(2.6, 2.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"22\",side=2,line=0,outer=FALSE,at=3.0)\n",
      "> start <- c(11385911, 15169820, 23791794, 40504155, 41190872, 99823014, 107463350, 118018473, 119619530, 151881865, 152992532)\n",
      "> end <- c(11385938, 15169885, 23791879, 40504219, 41191001, 99823041, 107463445, 118018508, 119619557, 151881938, 152992565)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 1.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(1.6, 1.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"X\",side=2,line=0,outer=FALSE,at=2.0)\n",
      "> start <- c(10087337, 13109751, 13689874, 28789170, 58819467)\n",
      "> end <- c(10087364, 13109820, 13689908, 28789223, 58819494)\n",
      "> vals <- c(1.0, 1.0, 1.0, 1.0, 1.0)\n",
      "> vals[vals > 1.0] <- 1.0\n",
      "> vals[vals < 0] <- 0\n",
      "> heights <- 0.800000 * ((vals - 0)/(1.0 - 0)) + 0.6\n",
      "> for (i in 1:length(heights)) {\n",
      "+ \tpolygon(x=c(start[i], end[i], end[i], start[i]), y=c(0.6, 0.6, heights[i], heights[i]), col=c(\"#CC0000\"), border=c(\"#CC0000\"))\n",
      "+ }\n",
      "> mtext(\"Y\",side=2,line=0,outer=FALSE,at=1.0)\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "INFO  @ Tue, 15 Jun 2021 08:19:02: #... cong! See STAT1_6h_IFNa_bed_CEAS.pdf for the graphical results of CEAS! \n"
     ]
    }
   ],
   "source": [
    "# the output folder is a STAT1_6h_IFNa_bed_CEAS.pdf file. \n",
    "\n",
    "ceas --name=STAT1_6h_IFNa_bed_CEAS -g /fdb/CEAS/hg19.refGene -b STAT1_6h_IFNa_peaks.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "447e665b-1820-4cf0-aec1-01bde1c54851",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[01;31m\u001b[KSTAT1_6h_IFNa_bed_CEAS\u001b[m\u001b[K.pdf\n",
      "\u001b[01;31m\u001b[KSTAT1_6h_IFNa_bed_CEAS\u001b[m\u001b[K.R\n",
      "\u001b[01;31m\u001b[KSTAT1_6h_IFNa_bed_CEAS\u001b[m\u001b[K.xls\n"
     ]
    }
   ],
   "source": [
    "# the output folder are three files\n",
    "ls | grep STAT1_6h_IFNa_bed_CEAS\n",
    "# xls, contain the nearest gene (in refseq ID and distance)\n",
    "# pdf is a summary"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f8ac70ab-755c-4e2b-9a3c-027c903312b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: write error: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "cat STAT1_6h_IFNa_bed_CEAS.xls | head -n 20 | tail -n 3 > ceas_title.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "5c1e60d0-fd5f-428b-860f-46384cef5af8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#name\tchr\ttxStart\ttxEnd\tstrand\tdist u TSS\tdist d TSS\tdist u TTS\tdist d TTS\t3000bp u TSS\t3000bp d TSS\t1/3 gene\t2/3 gene\t3/3 gene\t3000bp d TTS\texons\n",
      "NR_024540\tchr1\t14362\t29370\t-\t919308\tNA\t934316\tNA\t0\t0\t0\t0\t0\t0\t0\n",
      "NR_026818\tchr1\t34611\t36081\t-\t912597\tNA\t914067\tNA\t0\t0\t0\t0\t0\t0\t0\n"
     ]
    }
   ],
   "source": [
    "cat ceas_title.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5e1c4659-1793-4b52-896f-ff600431bd0e",
   "metadata": {},
   "source": [
    "# CEAS Promoter analysis of IFN-stimulated genes (ISGs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e24b91ae-ace8-478b-a82c-20e06f43664e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This ISGs.txt are a manual preparation from a separate source\n",
    "# ISGs.txt generated from supp table I in this paper https://www.pnas.org/content/109/11/4239"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12c3514a-02e3-455b-853c-5ce8b0df0a21",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this following step take a very long time (~10-15 minutes) to finish."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ca9139af-2522-4e45-b513-7e2977a85f0e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Tue, 15 Jun 2021 08:27:11: \n",
      "# ARGUMENTS: \n",
      "# name = STAT1_6h_IFNa_ISGs_CEAS\n",
      "# gene annotation table = /fdb/CEAS/hg19.refGene\n",
      "# BED file = None\n",
      "# WIG file = wig/STAT1_6h_IFNa.wig\n",
      "# extra BED file = None\n",
      "# ChIP annotation = Off\n",
      "# gene-centered annotation =  Off\n",
      "# average profiling = On\n",
      "# dump profiles = On\n",
      "# profiling resolution (average profiling) = 50 bp\n",
      "# relative distance wrt TSS and TTS (average profiling) = 3000 bp\n",
      "# gene groups (average profiling) = ISGs.txt \n",
      "INFO  @ Tue, 15 Jun 2021 08:27:11: #1 read the gene table... \n",
      "INFO  @ Tue, 15 Jun 2021 08:27:14: #2-1 run wig profiling of chr1... \n",
      "INFO  @ Tue, 15 Jun 2021 08:30:39: #2-2 run wig profiling of chr10... \n",
      "INFO  @ Tue, 15 Jun 2021 08:31:19: #2-3 run wig profiling of chr11... \n",
      "INFO  @ Tue, 15 Jun 2021 08:31:58: #2-4 run wig profiling of chr12... \n",
      "INFO  @ Tue, 15 Jun 2021 08:32:37: #2-5 run wig profiling of chr13... \n",
      "INFO  @ Tue, 15 Jun 2021 08:32:59: #2-6 run wig profiling of chr14... \n",
      "INFO  @ Tue, 15 Jun 2021 08:33:23: #2-7 run wig profiling of chr15... \n",
      "INFO  @ Tue, 15 Jun 2021 08:33:47: #2-8 run wig profiling of chr16... \n",
      "INFO  @ Tue, 15 Jun 2021 08:34:10: #2-9 run wig profiling of chr17... \n",
      "INFO  @ Tue, 15 Jun 2021 08:34:36: #2-10 run wig profiling of chr18... \n",
      "INFO  @ Tue, 15 Jun 2021 08:34:52: #2-11 run wig profiling of chr19... \n",
      "INFO  @ Tue, 15 Jun 2021 08:35:11: #2-12 run wig profiling of chr2... \n",
      "INFO  @ Tue, 15 Jun 2021 08:36:24: #2-13 run wig profiling of chr20... \n",
      "INFO  @ Tue, 15 Jun 2021 08:36:40: #2-14 run wig profiling of chr21... \n",
      "INFO  @ Tue, 15 Jun 2021 08:36:49: #2-15 run wig profiling of chr22... \n",
      "INFO  @ Tue, 15 Jun 2021 08:37:01: #2-16 run wig profiling of chr3... \n",
      "INFO  @ Tue, 15 Jun 2021 08:37:55: #2-17 run wig profiling of chr4... \n",
      "INFO  @ Tue, 15 Jun 2021 08:38:39: #2-18 run wig profiling of chr5... \n",
      "INFO  @ Tue, 15 Jun 2021 08:39:23: #2-19 run wig profiling of chr6... \n",
      "INFO  @ Tue, 15 Jun 2021 08:40:08: #2-20 run wig profiling of chr7... \n",
      "INFO  @ Tue, 15 Jun 2021 08:40:55: #2-21 run wig profiling of chr8... \n",
      "INFO  @ Tue, 15 Jun 2021 08:41:33: #2-22 run wig profiling of chr9... \n",
      "INFO  @ Tue, 15 Jun 2021 08:42:06: #2-23 run wig profiling of chrX... \n",
      "INFO  @ Tue, 15 Jun 2021 08:42:51: #2-24 run wig profiling of chrY... \n",
      "INFO  @ Tue, 15 Jun 2021 08:42:59: #3 write an R script of wig profiling... \n",
      "\n",
      "R version 4.0.5 (2021-03-31) -- \"Shake and Throw\"\n",
      "Copyright (C) 2021 The R Foundation for Statistical Computing\n",
      "Platform: x86_64-pc-linux-gnu (64-bit)\n",
      "\n",
      "R is free software and comes with ABSOLUTELY NO WARRANTY.\n",
      "You are welcome to redistribute it under certain conditions.\n",
      "Type 'license()' or 'licence()' for distribution details.\n",
      "\n",
      "  Natural language support but running in an English locale\n",
      "\n",
      "R is a collaborative project with many contributors.\n",
      "Type 'contributors()' for more information and\n",
      "'citation()' on how to cite R or R packages in publications.\n",
      "\n",
      "Type 'demo()' for some demos, 'help()' for on-line help, or\n",
      "'help.start()' for an HTML browser interface to help.\n",
      "Type 'q()' to quit R.\n",
      "\n",
      "> # ARGUMENTS: \n",
      "> # name = STAT1_6h_IFNa_ISGs_CEAS\n",
      "> # gene annotation table = /fdb/CEAS/hg19.refGene\n",
      "> # BED file = None\n",
      "> # WIG file = wig/STAT1_6h_IFNa.wig\n",
      "> # extra BED file = None\n",
      "> # ChIP annotation = Off\n",
      "> # gene-centered annotation =  Off\n",
      "> # average profiling = On\n",
      "> # dump profiles = On\n",
      "> # profiling resolution (average profiling) = 50 bp\n",
      "> # relative distance wrt TSS and TTS (average profiling) = 3000 bp\n",
      "> # gene groups (average profiling) = ISGs.txt\n",
      "> pdf(\"STAT1_6h_IFNa_ISGs_CEAS.pdf\",height=11.5,width=8.5)\n",
      "> par(mar=c(4, 4, 5, 3.8),oma=c(4, 2, 4, 2))\n",
      "> layout(matrix(c(1, 2, 3, 3, 4, 5), 3, 2, byrow = TRUE),widths=c(1, 1),heights=c(1, 1, 1))\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(-3000.000000,-2950.000000,-2900.000000,-2850.000000,-2800.000000,-2750.000000,-2700.000000,-2650.000000,-2600.000000,-2550.000000,-2500.000000,-2450.000000,-2400.000000,-2350.000000,-2300.000000,-2250.000000,-2200.000000,-2150.000000,-2100.000000,-2050.000000,-2000.000000,-1950.000000,-1900.000000,-1850.000000,-1800.000000,-1750.000000,-1700.000000,-1650.000000,-1600.000000,-1550.000000,-1500.000000,-1450.000000,-1400.000000,-1350.000000,-1300.000000,-1250.000000,-1200.000000,-1150.000000,-1100.000000,-1050.000000,-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000)\n",
      "> y<-c(0.070545,0.307455,0.289455,0.182909,0.171636,0.103636,0.076545,0.157818,0.110909,0.103455,0.080909,0.148364,0.112364,0.179818,0.170545,0.189455,0.268727,0.094000,0.074364,0.130545,0.124909,0.116182,0.169091,0.163818,0.172000,0.138727,0.145818,0.151818,0.164000,0.132000,0.143636,0.110545,0.127273,0.162727,0.135455,0.121273,0.109273,0.131636,0.191273,0.159818,0.152182,0.124545,0.114182,0.166000,0.196727,0.137818,0.213091,0.275636,0.274727,0.224727,0.269273,0.279091,0.352545,0.428909,0.745455,0.985818,1.696364,2.505091,3.386000,4.573091,5.078182,5.488727,4.669636,2.406182,1.310545,1.035273,1.122545,1.207636,1.026182,0.672000,0.557091,0.496909,0.227091,0.196545,0.235636,0.207818,0.233818,0.151455,0.242000,0.306182,0.313273,0.296182,0.291636,0.197273,0.217091,0.166909,0.362545,0.337636,0.300545,0.281091,0.254364,0.233091,0.179636,0.158909,0.125091,0.116727,0.126182,0.100909,0.098909,0.113273,0.148000,0.118909,0.075455,0.149818,0.201091,0.145091,0.123273,0.161636,0.147455,0.184545,0.195455,0.190364,0.124545,0.180364,0.139091,0.158727,0.116364,0.094727,0.129636,0.151455,0.119818)\n",
      "> plot(x, y,type=\"l\",main=\"Average Profiles near TSS\",xlab=\"Relative Distance (bp)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.063491,6.037600),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(-3000.000000,-2950.000000,-2900.000000,-2850.000000,-2800.000000,-2750.000000,-2700.000000,-2650.000000,-2600.000000,-2550.000000,-2500.000000,-2450.000000,-2400.000000,-2350.000000,-2300.000000,-2250.000000,-2200.000000,-2150.000000,-2100.000000,-2050.000000,-2000.000000,-1950.000000,-1900.000000,-1850.000000,-1800.000000,-1750.000000,-1700.000000,-1650.000000,-1600.000000,-1550.000000,-1500.000000,-1450.000000,-1400.000000,-1350.000000,-1300.000000,-1250.000000,-1200.000000,-1150.000000,-1100.000000,-1050.000000,-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000)\n",
      "> y<-c(0.117195,0.116385,0.116978,0.114799,0.113832,0.116561,0.114229,0.116560,0.117755,0.117887,0.115980,0.118252,0.117268,0.118033,0.120537,0.120890,0.116939,0.118697,0.119685,0.119721,0.122829,0.120547,0.119019,0.121422,0.123960,0.122931,0.124772,0.125211,0.123709,0.123761,0.123648,0.124274,0.123451,0.128621,0.128583,0.130984,0.129580,0.129996,0.129919,0.134184,0.135748,0.136233,0.139532,0.138235,0.143459,0.138544,0.142581,0.144653,0.148256,0.150672,0.154089,0.160902,0.164994,0.175439,0.185286,0.197592,0.217074,0.230932,0.239770,0.248457,0.246556,0.230050,0.196621,0.167711,0.151055,0.142629,0.135312,0.129059,0.130345,0.130614,0.129292,0.133396,0.135526,0.134161,0.136363,0.137908,0.134559,0.135728,0.138346,0.136330,0.133170,0.132197,0.129715,0.133664,0.134230,0.128856,0.128255,0.126975,0.125104,0.124885,0.125743,0.129676,0.122259,0.123911,0.126737,0.127512,0.121889,0.119153,0.121770,0.121990,0.120794,0.119290,0.116904,0.116844,0.116725,0.120340,0.117282,0.120545,0.118718,0.116999,0.117156,0.114351,0.116252,0.114105,0.114829,0.113828,0.114745,0.113780,0.111715,0.115970,0.113891)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> abline(v=0.000000,lty=2,col=c(\"black\"))\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(-3000.000000,-2950.000000,-2900.000000,-2850.000000,-2800.000000,-2750.000000,-2700.000000,-2650.000000,-2600.000000,-2550.000000,-2500.000000,-2450.000000,-2400.000000,-2350.000000,-2300.000000,-2250.000000,-2200.000000,-2150.000000,-2100.000000,-2050.000000,-2000.000000,-1950.000000,-1900.000000,-1850.000000,-1800.000000,-1750.000000,-1700.000000,-1650.000000,-1600.000000,-1550.000000,-1500.000000,-1450.000000,-1400.000000,-1350.000000,-1300.000000,-1250.000000,-1200.000000,-1150.000000,-1100.000000,-1050.000000,-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000)\n",
      "> y<-c(0.264364,0.284727,0.140909,0.152727,0.141273,0.206000,0.092182,0.141818,0.081273,0.096182,0.086727,0.058545,0.109273,0.071818,0.124545,0.063818,0.101636,0.122545,0.072364,0.120909,0.089818,0.127818,0.115273,0.072364,0.151455,0.125636,0.123455,0.138364,0.113091,0.104545,0.114182,0.076727,0.106909,0.152000,0.126000,0.099273,0.092909,0.057818,0.115091,0.151273,0.092727,0.105091,0.094000,0.091273,0.112364,0.129091,0.091455,0.129818,0.105091,0.072182,0.062364,0.071273,0.085818,0.126727,0.078909,0.109273,0.104727,0.118909,0.081455,0.106727,0.140909,0.113636,0.173273,0.214909,0.174000,0.108000,0.113636,0.120909,0.113091,0.138545,0.185818,0.186000,0.057818,0.113455,0.140909,0.156545,0.115455,0.195818,0.219273,0.144727,0.137455,0.160182,0.079818,0.099636,0.167273,0.113273,0.100727,0.121091,0.076364,0.140182,0.132364,0.135273,0.171273,0.121273,0.258182,0.294000,0.239273,0.196182,0.104364,0.169273,0.167818,0.162182,0.153091,0.104545,0.123818,0.110000,0.095455,0.085455,0.148909,0.140909,0.104909,0.089273,0.149455,0.110364,0.111273,0.121636,0.125455,0.127273,0.112182,0.114364,0.116364)\n",
      "> plot(x, y,type=\"l\",main=\"Average Profiles near TTS\",xlab=\"Relative Distance (bp)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.052036,0.323400),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(-3000.000000,-2950.000000,-2900.000000,-2850.000000,-2800.000000,-2750.000000,-2700.000000,-2650.000000,-2600.000000,-2550.000000,-2500.000000,-2450.000000,-2400.000000,-2350.000000,-2300.000000,-2250.000000,-2200.000000,-2150.000000,-2100.000000,-2050.000000,-2000.000000,-1950.000000,-1900.000000,-1850.000000,-1800.000000,-1750.000000,-1700.000000,-1650.000000,-1600.000000,-1550.000000,-1500.000000,-1450.000000,-1400.000000,-1350.000000,-1300.000000,-1250.000000,-1200.000000,-1150.000000,-1100.000000,-1050.000000,-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000)\n",
      "> y<-c(0.107348,0.108631,0.106731,0.105599,0.106968,0.107627,0.106622,0.103279,0.106777,0.101386,0.105248,0.103277,0.104827,0.107139,0.108115,0.102682,0.101490,0.103729,0.103717,0.109126,0.103566,0.104745,0.104915,0.105745,0.105532,0.107633,0.106156,0.106513,0.104363,0.104957,0.102972,0.107642,0.105684,0.104801,0.106028,0.101800,0.106624,0.105158,0.106169,0.107280,0.102447,0.103418,0.103174,0.104703,0.105818,0.105582,0.103110,0.102669,0.104040,0.102272,0.107902,0.106487,0.106916,0.105428,0.108523,0.107616,0.109946,0.109972,0.107684,0.109436,0.108573,0.107583,0.109724,0.111405,0.112010,0.115562,0.110973,0.114947,0.113903,0.112627,0.111152,0.110791,0.110335,0.113587,0.109742,0.113298,0.115375,0.114320,0.111750,0.115610,0.108955,0.108444,0.112237,0.113881,0.113939,0.112797,0.108261,0.112453,0.112958,0.112960,0.112271,0.110057,0.113124,0.114665,0.113584,0.116209,0.115952,0.111468,0.112299,0.114046,0.115869,0.118264,0.118278,0.117370,0.116242,0.111090,0.110408,0.111089,0.109675,0.110362,0.112033,0.110080,0.111595,0.112634,0.108103,0.111889,0.111131,0.111119,0.114878,0.108855,0.110882)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> abline(v=0.000000,lty=2,col=c(\"black\"))\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000,3050.000000,3100.000000,3150.000000,3200.000000,3250.000000,3300.000000,3350.000000,3400.000000,3450.000000,3500.000000,3550.000000,3600.000000,3650.000000,3700.000000,3750.000000,3800.000000,3850.000000,3900.000000,3950.000000,4000.000000)\n",
      "> y<-c(0.152182,0.124545,0.114182,0.166000,0.196727,0.137818,0.213091,0.275636,0.274727,0.224727,0.269273,0.279091,0.352545,0.428909,0.745455,0.985818,1.696364,2.505091,3.386000,4.573091,5.078182,2.056182,0.687273,0.371818,0.399636,0.421818,0.478909,0.394000,0.338727,0.338909,0.189818,0.179818,0.165091,0.094727,0.138909,0.135273,0.158727,0.092364,0.150727,0.107636,0.157636,0.130000,0.282182,0.132727,0.108182,0.127091,0.160727,0.130364,0.086364,0.104909,0.136545,0.159636,0.108909,0.111273,0.124545,0.089818,0.160545,0.099091,0.105455,0.183455,0.074364,0.147455,0.123636,0.132545,0.071455,0.121818,0.140364,0.118000,0.088545,0.187818,0.112545,0.082545,0.086727,0.099455,0.102000,0.146909,0.088182,0.101273,0.136909,0.153455,0.096909,0.104545,0.113636,0.173273,0.214909,0.174000,0.108000,0.113636,0.120909,0.113091,0.138545,0.185818,0.186000,0.057818,0.113455,0.140909,0.156545,0.115455,0.195818,0.219273,0.144727,0.137455)\n",
      "> plot(x, y,type=\"l\",main=\"Average Gene Profiles\",xlab=\"Upstream (bp), 3000 bp of Meta-gene, Downstream (bp)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.052036,5.586000),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(-1000.000000,-950.000000,-900.000000,-850.000000,-800.000000,-750.000000,-700.000000,-650.000000,-600.000000,-550.000000,-500.000000,-450.000000,-400.000000,-350.000000,-300.000000,-250.000000,-200.000000,-150.000000,-100.000000,-50.000000,0.000000,0.000000,50.000000,100.000000,150.000000,200.000000,250.000000,300.000000,350.000000,400.000000,450.000000,500.000000,550.000000,600.000000,650.000000,700.000000,750.000000,800.000000,850.000000,900.000000,950.000000,1000.000000,1050.000000,1100.000000,1150.000000,1200.000000,1250.000000,1300.000000,1350.000000,1400.000000,1450.000000,1500.000000,1550.000000,1600.000000,1650.000000,1700.000000,1750.000000,1800.000000,1850.000000,1900.000000,1950.000000,2000.000000,2050.000000,2100.000000,2150.000000,2200.000000,2250.000000,2300.000000,2350.000000,2400.000000,2450.000000,2500.000000,2550.000000,2600.000000,2650.000000,2700.000000,2750.000000,2800.000000,2850.000000,2900.000000,2950.000000,3000.000000,3050.000000,3100.000000,3150.000000,3200.000000,3250.000000,3300.000000,3350.000000,3400.000000,3450.000000,3500.000000,3550.000000,3600.000000,3650.000000,3700.000000,3750.000000,3800.000000,3850.000000,3900.000000,3950.000000,4000.000000)\n",
      "> y<-c(0.135748,0.136233,0.139532,0.138235,0.143459,0.138544,0.142581,0.144653,0.148256,0.150672,0.154089,0.160902,0.164994,0.175439,0.185286,0.197592,0.217074,0.230932,0.239770,0.248457,0.246556,0.188287,0.154967,0.150365,0.143340,0.142470,0.138365,0.145314,0.143280,0.143110,0.138031,0.137768,0.139659,0.137823,0.139872,0.136055,0.137013,0.141733,0.139503,0.140275,0.139229,0.133961,0.140937,0.137503,0.138117,0.140918,0.138417,0.140300,0.138345,0.139022,0.141493,0.140836,0.140218,0.139525,0.138955,0.135838,0.139129,0.141055,0.139345,0.139757,0.138108,0.139745,0.135461,0.137695,0.139354,0.139514,0.142354,0.142122,0.141546,0.141706,0.144153,0.142806,0.139461,0.141816,0.139218,0.140822,0.145415,0.141875,0.142285,0.139137,0.140920,0.142275,0.107583,0.109724,0.111405,0.112010,0.115562,0.110973,0.114947,0.113903,0.112627,0.111152,0.110791,0.110335,0.113587,0.109742,0.113298,0.115375,0.114320,0.111750,0.115610,0.108955)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> abline(v=0.000000,lty=2,col=c(\"black\"))\n",
      "> abline(v=3000.000000,lty=2,col=c(\"black\"))\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(5.396545,2.582545,0.920000,0.408727,0.231818,0.178182,0.142000,0.176545,0.118545,0.123091,0.148727,0.125455,0.130909,0.118727,0.129091,0.147636,0.120000,0.124000,0.143818,0.091091,0.112727,0.107636,0.111818,0.173636,0.116000,0.122545,0.140909,0.129818,0.112727,0.112364,0.180545)\n",
      "> plot(x, y,type=\"l\",main=\"Average Concatenated Exon Profiles\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.069818,6.475855),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(0.232140,0.185917,0.156836,0.149513,0.143973,0.139451,0.140323,0.140761,0.140455,0.141286,0.141272,0.143629,0.144834,0.144906,0.148104,0.148506,0.148514,0.150145,0.149678,0.149032,0.154535,0.155288,0.157296,0.158782,0.160591,0.159107,0.161189,0.164518,0.160535,0.160105,0.162535)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(0.370909,0.275273,0.367818,0.415636,0.386727,0.132364,0.099636,0.106545,0.102000,0.133091,0.118182,0.119455,0.207636,0.162182,0.140364,0.088909,0.094545,0.113455,0.128364,0.139636,0.112000,0.111091,0.077273,0.100000,0.138000,0.112909,0.120727,0.125636,0.069818,0.127273,0.177091)\n",
      "> plot(x, y,type=\"l\",main=\"Average Concatenated Intron Profiles\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.069818,6.475855),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(0.138096,0.126975,0.124845,0.119462,0.120017,0.119109,0.116659,0.118523,0.115376,0.116774,0.115449,0.116091,0.114587,0.115170,0.117001,0.113559,0.115170,0.113900,0.114856,0.115155,0.118359,0.115718,0.116343,0.116528,0.116171,0.115277,0.111640,0.112212,0.113071,0.113221,0.111487)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> par(mfrow=c(3, 2),mar=c(4, 4, 5, 3.8),oma=c(4, 2, 4, 2))\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,50.000000,100.000000)\n",
      "> y<-c(0.189507,0.196502,0.170314)\n",
      "> plot(x, y,type=\"l\",main=\"Average Exon Profiles\n",
      "+ (66 <= length < 109 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,50.000000,100.000000)\n",
      "> y<-c(0.104275,0.104505,0.148103)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,12.500000,25.000000,37.500000,50.000000,62.500000,75.000000,87.500000,100.000000)\n",
      "> y<-c(0.195020,0.191807,0.177189,0.224418,0.194458,0.281767,0.329880,0.338153,0.319277)\n",
      "> plot(x, y,type=\"l\",main=\"Average Intron Profiles\n",
      "+ (151 <= length < 812 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,12.500000,25.000000,37.500000,50.000000,62.500000,75.000000,87.500000,100.000000)\n",
      "> y<-c(0.097108,0.105479,0.120300,0.135203,0.144509,0.157162,0.162515,0.173012,0.182147)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,50.000000,100.000000)\n",
      "> y<-c(0.313710,0.345300,0.221908)\n",
      "> plot(x, y,type=\"l\",main=\"Average Exon Profiles\n",
      "+ (109 <= length < 165 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,50.000000,100.000000)\n",
      "> y<-c(0.109913,0.106689,0.100854)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(0.174821,0.174702,0.203571,0.175952,0.202560,0.207560,0.214286,0.185714,0.180833,0.193214,0.245238,0.226964,0.206905,0.236667,0.238750,0.268929,0.235595,0.255774,0.276131,0.285417,0.261845,0.268393,0.273155,0.278690,0.276786,0.262143,0.259048,0.275238,0.308810,0.287321,0.287143)\n",
      "> plot(x, y,type=\"l\",main=\"Average Intron Profiles\n",
      "+ (812 <= length < 2687 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,3.333333,6.666667,10.000000,13.333333,16.666667,20.000000,23.333333,26.666667,30.000000,33.333333,36.666667,40.000000,43.333333,46.666667,50.000000,53.333333,56.666667,60.000000,63.333333,66.666667,70.000000,73.333333,76.666667,80.000000,83.333333,86.666667,90.000000,93.333333,96.666667,100.000000)\n",
      "> y<-c(0.101335,0.111763,0.132367,0.145873,0.160696,0.175548,0.186493,0.193105,0.204992,0.209599,0.216461,0.221799,0.226015,0.231967,0.235805,0.238255,0.240911,0.242676,0.245053,0.247835,0.251654,0.253832,0.256565,0.256048,0.257113,0.256233,0.256904,0.255094,0.253282,0.254435,0.258104)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,25.000000,50.000000,75.000000,100.000000)\n",
      "> y<-c(1.246860,1.170349,0.864244,0.412267,0.327965)\n",
      "> plot(x, y,type=\"l\",main=\"Average Exon Profiles\n",
      "+ (165 <= length < 501 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,25.000000,50.000000,75.000000,100.000000)\n",
      "> y<-c(0.130051,0.151566,0.144196,0.155533,0.163758)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> cr <- colorRampPalette(col=c(\"#C8524D\", \"#BDD791\", \"#447CBE\", \"#775A9C\"), bias=1)\n",
      "> linecols <- cr(1)\n",
      "> linecols <- c(linecols, \"black\")\n",
      "> x<-c(0.000000,1.063830,2.127660,3.191489,4.255319,5.319149,6.382979,7.446809,8.510638,9.574468,10.638298,11.702128,12.765957,13.829787,14.893617,15.957447,17.021277,18.085106,19.148936,20.212766,21.276596,22.340426,23.404255,24.468085,25.531915,26.595745,27.659574,28.723404,29.787234,30.851064,31.914894,32.978723,34.042553,35.106383,36.170213,37.234043,38.297872,39.361702,40.425532,41.489362,42.553191,43.617021,44.680851,45.744681,46.808511,47.872340,48.936170,50.000000,51.063830,52.127660,53.191489,54.255319,55.319149,56.382979,57.446809,58.510638,59.574468,60.638298,61.702128,62.765957,63.829787,64.893617,65.957447,67.021277,68.085106,69.148936,70.212766,71.276596,72.340426,73.404255,74.468085,75.531915,76.595745,77.659574,78.723404,79.787234,80.851064,81.914894,82.978723,84.042553,85.106383,86.170213,87.234043,88.297872,89.361702,90.425532,91.489362,92.553191,93.617021,94.680851,95.744681,96.808511,97.872340,98.936170,100.000000)\n",
      "> y<-c(0.449695,0.402672,0.390229,0.474504,0.444351,0.376412,0.300000,0.291985,0.271985,0.252519,0.240534,0.327176,0.281908,0.269466,0.244580,0.238473,0.261298,0.291832,0.275344,0.269847,0.292977,0.307557,0.266336,0.286565,0.295038,0.242672,0.251603,0.278244,0.266336,0.270229,0.298702,0.288397,0.295878,0.279389,0.298779,0.263053,0.258779,0.263130,0.293511,0.342824,0.272214,0.313282,0.286107,0.258244,0.282519,0.267481,0.268855,0.267786,0.240153,0.255802,0.280763,0.256412,0.259695,0.260229,0.279924,0.269237,0.244962,0.264351,0.274351,0.300840,0.264275,0.268931,0.294885,0.270916,0.248244,0.281985,0.294885,0.312672,0.348626,0.354427,0.413588,0.450076,0.347023,0.307099,0.266489,0.239008,0.276107,0.280992,0.267023,0.274962,0.276107,0.281298,0.303511,0.267252,0.290458,0.260305,0.247328,0.277863,0.251450,0.235267,0.216794,0.254809,0.249618,0.234198,0.282595)\n",
      "> plot(x, y,type=\"l\",main=\"Average Intron Profiles\n",
      "+ (2687 <= length < 11850 bp)\",xlab=\"Relative Location (%)\",ylab=\"Average Profile\",col=linecols[1],ylim=c(0.097108,1.496233),xaxt=\"s\",yaxt=\"s\",lwd=2)\n",
      "> x <- c(0.000000,1.063830,2.127660,3.191489,4.255319,5.319149,6.382979,7.446809,8.510638,9.574468,10.638298,11.702128,12.765957,13.829787,14.893617,15.957447,17.021277,18.085106,19.148936,20.212766,21.276596,22.340426,23.404255,24.468085,25.531915,26.595745,27.659574,28.723404,29.787234,30.851064,31.914894,32.978723,34.042553,35.106383,36.170213,37.234043,38.297872,39.361702,40.425532,41.489362,42.553191,43.617021,44.680851,45.744681,46.808511,47.872340,48.936170,50.000000,51.063830,52.127660,53.191489,54.255319,55.319149,56.382979,57.446809,58.510638,59.574468,60.638298,61.702128,62.765957,63.829787,64.893617,65.957447,67.021277,68.085106,69.148936,70.212766,71.276596,72.340426,73.404255,74.468085,75.531915,76.595745,77.659574,78.723404,79.787234,80.851064,81.914894,82.978723,84.042553,85.106383,86.170213,87.234043,88.297872,89.361702,90.425532,91.489362,92.553191,93.617021,94.680851,95.744681,96.808511,97.872340,98.936170,100.000000)\n",
      "> y<-c(0.106623,0.118072,0.136136,0.153887,0.167384,0.179981,0.191764,0.201190,0.212664,0.217338,0.223875,0.228274,0.233250,0.236716,0.241512,0.245727,0.247543,0.251019,0.252990,0.256325,0.256645,0.259149,0.260084,0.260616,0.261894,0.262270,0.263228,0.262934,0.264500,0.264887,0.265673,0.265586,0.264648,0.264510,0.265671,0.268567,0.267319,0.267224,0.266053,0.266569,0.267079,0.267241,0.268062,0.269340,0.269464,0.268537,0.268519,0.268263,0.267556,0.267810,0.267014,0.266802,0.267837,0.269034,0.269542,0.270024,0.268186,0.270039,0.268927,0.269618,0.265952,0.269619,0.267984,0.267437,0.267162,0.268192,0.267101,0.268366,0.267654,0.267445,0.267928,0.269063,0.270430,0.268160,0.268241,0.269720,0.270755,0.269170,0.269390,0.269666,0.267740,0.268341,0.268936,0.266971,0.267656,0.268134,0.268242,0.267889,0.267124,0.267878,0.266734,0.266063,0.265275,0.264016,0.266987)\n",
      "> lines(x, y,type=\"l\",col=linecols[2],lwd=2)\n",
      "> legend(\"topleft\",legend=c(\"ISGs\",\"All\"),col=linecols,pch=15,bty=\"o\")\n",
      "> dev.off()\n",
      "null device \n",
      "          1 \n",
      "> \n",
      "INFO  @ Tue, 15 Jun 2021 08:43:00: #... cong! See STAT1_6h_IFNa_ISGs_CEAS.pdf for the graphical results of CEAS! \n"
     ]
    }
   ],
   "source": [
    "ceas --name=STAT1_6h_IFNa_ISGs_CEAS --pf-res=50 -g /fdb/CEAS/hg19.refGene --dump --gn-groups=ISGs.txt --gn-group-names='ISGs' -w wig/STAT1_6h_IFNa.wig"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39b29fc4-2806-4d1a-9d3d-ce0dd11e67bd",
   "metadata": {},
   "source": [
    "# use Homer to find enriched Motifs in Genomic Regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "45ba24fb-a490-4616-8817-f5045f804e06",
   "metadata": {},
   "outputs": [],
   "source": [
    "# finding motif takes >30 minutes to finish for each sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "884e4a1d-e7d3-4895-b905-52a13faac842",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPosition file = peak/STAT1_6h_IFNa_summits.bed\n",
      "\tGenome = hg19\n",
      "\tOutput Directory = STAT1_6h_IFNa_homer\n",
      "\tFragment size set to 200\n",
      "\tFound mset for \"human\", will check against vertebrates motifs\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 1085\n",
      "\t\tpeakfile formatted lines: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 1085\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tBackground files for 200 bp fragments found.\n",
      "\tCustom genome sequence directory: /usr/local/apps/homer/4.11.1/.//data/genomes/hg19//\n",
      "\n",
      "\tExtracting sequences from directory: /usr/local/apps/homer/4.11.1/.//data/genomes/hg19//\n",
      "\tExtracting 56 sequences from chr1\n",
      "\tExtracting 64 sequences from chr2\n",
      "\tExtracting 29 sequences from chr3\n",
      "\tExtracting 43 sequences from chr4\n",
      "\tExtracting 26 sequences from chr5\n",
      "\tExtracting 52 sequences from chr6\n",
      "\tExtracting 39 sequences from chr7\n",
      "\tExtracting 18 sequences from chr8\n",
      "\tExtracting 25 sequences from chr9\n",
      "\tExtracting 43 sequences from chr10\n",
      "\tExtracting 39 sequences from chr11\n",
      "\tExtracting 37 sequences from chr12\n",
      "\tExtracting 10 sequences from chr13\n",
      "\tExtracting 16 sequences from chr14\n",
      "\tExtracting 21 sequences from chr15\n",
      "\tExtracting 23 sequences from chr16\n",
      "\tExtracting 40 sequences from chr17\n",
      "\tExtracting 8 sequences from chr18\n",
      "\tExtracting 17 sequences from chr19\n",
      "\tExtracting 14 sequences from chr20\n",
      "\tExtracting 26 sequences from chr21\n",
      "\tExtracting 23 sequences from chr22\n",
      "\tExtracting 18 sequences from chrX\n",
      "\tExtracting 398 sequences from chrY\n",
      "\n",
      "\tNot removing redundant sequences\n",
      "\n",
      "\n",
      "\tSequences processed:\n",
      "\t\tAuto detected maximum sequence length of 201 bp\n",
      "\t\t1085 total\n",
      "\n",
      "\tFrequency Bins: 0.2 0.25 0.3 0.35 0.4 0.45 0.5 0.6 0.7 0.8\n",
      "\tFreq\tBin\tCount\n",
      "\t0.3\t2\t6\n",
      "\t0.35\t3\t58\n",
      "\t0.4\t4\t374\n",
      "\t0.45\t5\t290\n",
      "\t0.5\t6\t158\n",
      "\t0.6\t7\t136\n",
      "\t0.7\t8\t40\n",
      "\t0.8\t9\t22\n",
      "\t10\t10\t1\n",
      "\n",
      "\tTotal sequences set to 50000\n",
      "\n",
      "\tChoosing background that matches in CpG/GC content...\n",
      "\tBin\t# Targets\t# Background\tBackground Weight\n",
      "\t2\t6\t270\t1.002\n",
      "\t3\t58\t2615\t1.000\n",
      "\t4\t374\t16861\t1.000\n",
      "\t5\t290\t13074\t1.000\n",
      "\t6\t158\t7123\t1.000\n",
      "\t7\t136\t6131\t1.000\n",
      "\t8\t40\t1803\t1.000\n",
      "\t9\t22\t992\t1.000\n",
      "\t10\t1\t45\t1.002\n",
      "\tAssembling sequence file...\n",
      "\tNormalizing lower order oligos using homer2\n",
      "\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t28.03%\t28.06%\t0.999\n",
      "\t\tC\t21.97%\t21.94%\t1.001\n",
      "\t\tG\t21.97%\t21.94%\t1.001\n",
      "\t\tT\t28.03%\t28.06%\t0.999\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t9.14%\t8.96%\t1.020\n",
      "\t\tCA\t7.30%\t7.33%\t0.997\n",
      "\t\tGA\t7.50%\t6.11%\t1.228\n",
      "\t\tTA\t4.08%\t5.66%\t0.721\n",
      "\t\tAC\t4.17%\t5.11%\t0.817\n",
      "\t\tCC\t6.47%\t5.91%\t1.095\n",
      "\t\tGC\t3.82%\t4.82%\t0.793\n",
      "\t\tTC\t7.50%\t6.11%\t1.228\n",
      "\t\tAG\t5.78%\t7.26%\t0.796\n",
      "\t\tCG\t2.42%\t1.45%\t1.667\n",
      "\t\tGG\t6.47%\t5.91%\t1.095\n",
      "\t\tTG\t7.30%\t7.33%\t0.997\n",
      "\t\tAT\t8.93%\t6.73%\t1.328\n",
      "\t\tCT\t5.78%\t7.26%\t0.796\n",
      "\t\tGT\t4.17%\t5.11%\t0.817\n",
      "\t\tTT\t9.14%\t8.96%\t1.020\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tNormalization weights can be found in file: STAT1_6h_IFNa_homer/seq.autonorm.tsv\n",
      "\tConverging on autonormalization solution:\n",
      "\t...............................................................................\n",
      "\tFinal normalization:\tAutonormalization: 1-mers (4 total)\n",
      "\t\tA\t28.03%\t27.96%\t1.002\n",
      "\t\tC\t21.97%\t22.04%\t0.997\n",
      "\t\tG\t21.97%\t22.04%\t0.997\n",
      "\t\tT\t28.03%\t27.96%\t1.002\n",
      "\tAutonormalization: 2-mers (16 total)\n",
      "\t\tAA\t9.14%\t8.99%\t1.017\n",
      "\t\tCA\t7.30%\t7.21%\t1.014\n",
      "\t\tGA\t7.50%\t6.61%\t1.134\n",
      "\t\tTA\t4.08%\t5.15%\t0.792\n",
      "\t\tAC\t4.17%\t4.75%\t0.878\n",
      "\t\tCC\t6.47%\t6.17%\t1.048\n",
      "\t\tGC\t3.82%\t4.49%\t0.851\n",
      "\t\tTC\t7.50%\t6.61%\t1.134\n",
      "\t\tAG\t5.78%\t6.79%\t0.851\n",
      "\t\tCG\t2.42%\t1.87%\t1.294\n",
      "\t\tGG\t6.47%\t6.17%\t1.048\n",
      "\t\tTG\t7.30%\t7.21%\t1.014\n",
      "\t\tAT\t8.93%\t7.43%\t1.201\n",
      "\t\tCT\t5.78%\t6.79%\t0.851\n",
      "\t\tGT\t4.17%\t4.75%\t0.878\n",
      "\t\tTT\t9.14%\t8.99%\t1.017\n",
      "\tAutonormalization: 3-mers (64 total)\n",
      "\tFinished preparing sequence/group files\n",
      "\n",
      "\t----------------------------------------------------------\n",
      "\tKnown motif enrichment\n",
      "\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\t440 motifs loaded\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 440 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tPreparing HTML output with sequence logos...\n",
      "\t\t1 of 440 (1e-327) ISRE(IRF)/ThioMac-LPS-Expression(GSE23622)/Homer\n",
      "\t\t2 of 440 (1e-317) IRF2(IRF)/Erythroblas-IRF2-ChIP-Seq(GSE36985)/Homer\n",
      "\t\t3 of 440 (1e-316) IRF3(IRF)/BMDM-Irf3-ChIP-Seq(GSE67343)/Homer\n",
      "\t\t4 of 440 (1e-285) IRF8(IRF)/BMDM-IRF8-ChIP-Seq(GSE77884)/Homer\n",
      "\t\t5 of 440 (1e-230) IRF1(IRF)/PBMC-IRF1-ChIP-Seq(GSE43036)/Homer\n",
      "\t\t6 of 440 (1e-222) ZNF189(Zf)/HEK293-ZNF189.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t\t7 of 440 (1e-126) PU.1:IRF8(ETS:IRF)/pDC-Irf8-ChIP-Seq(GSE66899)/Homer\n",
      "\t\t8 of 440 (1e-78) IRF4(IRF)/GM12878-IRF4-ChIP-Seq(GSE32465)/Homer\n",
      "\t\t9 of 440 (1e-77) TEAD3(TEA)/HepG2-TEAD3-ChIP-Seq(Encode)/Homer\n",
      "\t\t10 of 440 (1e-74) PU.1-IRF(ETS:IRF)/Bcell-PU.1-ChIP-Seq(GSE21512)/Homer\n",
      "\t\t11 of 440 (1e-72) TEAD(TEA)/Fibroblast-PU.1-ChIP-Seq(Unpublished)/Homer\n",
      "\t\t12 of 440 (1e-71) TEAD4(TEA)/Tropoblast-Tead4-ChIP-Seq(GSE37350)/Homer\n",
      "\t\t13 of 440 (1e-69) TEAD1(TEAD)/HepG2-TEAD1-ChIP-Seq(Encode)/Homer\n",
      "\t\t14 of 440 (1e-55) NFAT(RHD)/Jurkat-NFATC1-ChIP-Seq(Jolma_et_al.)/Homer\n",
      "\t\t15 of 440 (1e-54) HRE(HSF)/Striatum-HSF1-ChIP-Seq(GSE38000)/Homer\n",
      "\t\t16 of 440 (1e-52) T1ISRE(IRF)/ThioMac-Ifnb-Expression/Homer\n",
      "\t\t17 of 440 (1e-28) Hoxa10(Homeobox)/ChickenMSG-Hoxa10.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t18 of 440 (1e-22) PRDM1(Zf)/Hela-PRDM1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t19 of 440 (1e-19) Isl1(Homeobox)/Neuron-Isl1-ChIP-Seq(GSE31456)/Homer\n",
      "\t\t20 of 440 (1e-15) HRE(HSF)/HepG2-HSF1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t21 of 440 (1e-12) bZIP:IRF(bZIP,IRF)/Th17-BatF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t22 of 440 (1e-11) Six4(Homeobox)/MCF7-SIX4-ChIP-Seq(Encode)/Homer\n",
      "\t\t23 of 440 (1e-11) Bapx1(Homeobox)/VertebralCol-Bapx1-ChIP-Seq(GSE36672)/Homer\n",
      "\t\t24 of 440 (1e-10) IRF:BATF(IRF:bZIP)/pDC-Irf8-ChIP-Seq(GSE66899)/Homer\n",
      "\t\t25 of 440 (1e-10) Nkx2.2(Homeobox)/NPC-Nkx2.2-ChIP-Seq(GSE61673)/Homer\n",
      "\t\t26 of 440 (1e-10) Srebp1a(bHLH)/HepG2-Srebp1a-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t27 of 440 (1e-10) STAT5(Stat)/mCD4+-Stat5-ChIP-Seq(GSE12346)/Homer\n",
      "\t\t28 of 440 (1e-9) Gata1(Zf)/K562-GATA1-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t29 of 440 (1e-8) Bach1(bZIP)/K562-Bach1-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t30 of 440 (1e-8) CEBP:CEBP(bZIP)/MEF-Chop-ChIP-Seq(GSE35681)/Homer\n",
      "\t\t31 of 440 (1e-8) Hoxd10(Homeobox)/ChickenMSG-Hoxd10.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t32 of 440 (1e-8) Gata6(Zf)/HUG1N-GATA6-ChIP-Seq(GSE51936)/Homer\n",
      "\t\t33 of 440 (1e-7) Gata2(Zf)/K562-GATA2-ChIP-Seq(GSE18829)/Homer\n",
      "\t\t34 of 440 (1e-7) Nrf2(bZIP)/Lymphoblast-Nrf2-ChIP-Seq(GSE37589)/Homer\n",
      "\t\t35 of 440 (1e-7) Nkx2.5(Homeobox)/HL1-Nkx2.5.biotin-ChIP-Seq(GSE21529)/Homer\n",
      "\t\t36 of 440 (1e-7) NF-E2(bZIP)/K562-NFE2-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t37 of 440 (1e-6) Nkx6.1(Homeobox)/Islet-Nkx6.1-ChIP-Seq(GSE40975)/Homer\n",
      "\t\t38 of 440 (1e-6) Prop1(Homeobox)/GHFT1-PROP1.biotin-ChIP-Seq(GSE77302)/Homer\n",
      "\t\t39 of 440 (1e-6) CArG(MADS)/PUER-Srf-ChIP-Seq(Sullivan_et_al.)/Homer\n",
      "\t\t40 of 440 (1e-6) Sp5(Zf)/mES-Sp5.Flag-ChIP-Seq(GSE72989)/Homer\n",
      "\t\t41 of 440 (1e-6) Stat3+il21(Stat)/CD4-Stat3-ChIP-Seq(GSE19198)/Homer\n",
      "\t\t42 of 440 (1e-6) STAT1(Stat)/HelaS3-STAT1-ChIP-Seq(GSE12782)/Homer\n",
      "\t\t43 of 440 (1e-5) NFE2L2(bZIP)/HepG2-NFE2L2-ChIP-Seq(Encode)/Homer\n",
      "\t\t44 of 440 (1e-5) STAT4(Stat)/CD4-Stat4-ChIP-Seq(GSE22104)/Homer\n",
      "\t\t45 of 440 (1e-5) Sp1(Zf)/Promoter/Homer\n",
      "\t\t46 of 440 (1e-5) Sp2(Zf)/HEK293-Sp2.eGFP-ChIP-Seq(Encode)/Homer\n",
      "\t\t47 of 440 (1e-4) Gata4(Zf)/Heart-Gata4-ChIP-Seq(GSE35151)/Homer\n",
      "\t\t48 of 440 (1e-4) TRPS1(Zf)/MCF7-TRPS1-ChIP-Seq(GSE107013)/Homer\n",
      "\t\t49 of 440 (1e-4) Esrrb(NR)/mES-Esrrb-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t50 of 440 (1e-4) KLF3(Zf)/MEF-Klf3-ChIP-Seq(GSE44748)/Homer\n",
      "\t\t51 of 440 (1e-4) GATA:SCL(Zf,bHLH)/Ter119-SCL-ChIP-Seq(GSE18720)/Homer\n",
      "\t\t52 of 440 (1e-4) Fosl2(bZIP)/3T3L1-Fosl2-ChIP-Seq(GSE56872)/Homer\n",
      "\t\t53 of 440 (1e-4) MITF(bHLH)/MastCells-MITF-ChIP-Seq(GSE48085)/Homer\n",
      "\t\t54 of 440 (1e-3) Jun-AP1(bZIP)/K562-cJun-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t55 of 440 (1e-3) Arnt:Ahr(bHLH)/MCF7-Arnt-ChIP-Seq(Lo_et_al.)/Homer\n",
      "\t\t56 of 440 (1e-3) KLF1(Zf)/HUDEP2-KLF1-CutnRun(GSE136251)/Homer\n",
      "\t\t57 of 440 (1e-3) KLF5(Zf)/LoVo-KLF5-ChIP-Seq(GSE49402)/Homer\n",
      "\t\t58 of 440 (1e-3) RBPJ:Ebox(?,bHLH)/Panc1-Rbpj1-ChIP-Seq(GSE47459)/Homer\n",
      "\t\t59 of 440 (1e-3) Usf2(bHLH)/C2C12-Usf2-ChIP-Seq(GSE36030)/Homer\n",
      "\t\t60 of 440 (1e-3) BATF(bZIP)/Th17-BATF-ChIP-Seq(GSE39756)/Homer\n",
      "\t\t61 of 440 (1e-3) Hoxd12(Homeobox)/ChickenMSG-Hoxd12.Flag-ChIP-Seq(GSE86088)/Homer\n",
      "\t\t62 of 440 (1e-3) Brn1(POU,Homeobox)/NPC-Brn1-ChIP-Seq(GSE35496)/Homer\n",
      "\t\t63 of 440 (1e-3) Fli1(ETS)/CD8-FLI-ChIP-Seq(GSE20898)/Homer\n",
      "\t\t64 of 440 (1e-2) Klf9(Zf)/GBM-Klf9-ChIP-Seq(GSE62211)/Homer\n",
      "\t\t65 of 440 (1e-2) Bach2(bZIP)/OCILy7-Bach2-ChIP-Seq(GSE44420)/Homer\n",
      "\t\t66 of 440 (1e-2) Stat3(Stat)/mES-Stat3-ChIP-Seq(GSE11431)/Homer\n",
      "\t\t67 of 440 (1e-2) Elk4(ETS)/Hela-Elk4-ChIP-Seq(GSE31477)/Homer\n",
      "\t\t68 of 440 (1e-2) ERRg(NR)/Kidney-ESRRG-ChIP-Seq(GSE104905)/Homer\n",
      "\t\t69 of 440 (1e-2) CTCF(Zf)/CD4+-CTCF-ChIP-Seq(Barski_et_al.)/Homer\n",
      "\t\t70 of 440 (1e-2) ETS(ETS)/Promoter/Homer\n",
      "\t\t71 of 440 (1e-2) KLF6(Zf)/PDAC-KLF6-ChIP-Seq(GSE64557)/Homer\n",
      "\t\t72 of 440 (1e-2) MafK(bZIP)/C2C12-MafK-ChIP-Seq(GSE36030)/Homer\n",
      "\t\t73 of 440 (1e-2) Fra2(bZIP)/Striatum-Fra2-ChIP-Seq(GSE43429)/Homer\n",
      "\t\t74 of 440 (1e-2) Nr5a2(NR)/mES-Nr5a2-ChIP-Seq(GSE19019)/Homer\n",
      "\t\t75 of 440 (1e-2) MafB(bZIP)/BMM-Mafb-ChIP-Seq(GSE75722)/Homer\n",
      "\t\t76 of 440 (1e-2) JunB(bZIP)/DendriticCells-Junb-ChIP-Seq(GSE36099)/Homer\n",
      "\t\t77 of 440 (1e-2) Tlx?(NR)/NPC-H3K4me1-ChIP-Seq(GSE16256)/Homer\n",
      "\t\t78 of 440 (1e-2) ZSCAN22(Zf)/HEK293-ZSCAN22.GFP-ChIP-Seq(GSE58341)/Homer\n",
      "\t----------------------------------------------------------\n",
      "\tDe novo motif finding (HOMER)\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 32895\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 32895 of 34497 max\n",
      "\tTree  : 66860 of 172485 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 32895 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t17.25% skipped, 82.75% checked (27222 of 32895), of those checked:\n",
      "\t\t17.25% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 32895 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t17.25% skipped, 82.75% checked (27222 of 32895), of those checked:\n",
      "\t\t0.00% not in target, 59.34% increased p-value, 63.68% high p-value\n",
      "\n",
      "\tScreening oligos 32895 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t86.23% skipped, 13.77% checked (4528 of 32895), of those checked:\n",
      "\t\t0.00% not in target, 7.54% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: CCATTCCA... (-3764.754)\n",
      "\t\tRound 1: -3764.75 CCATTCCA T:2746.0(92.05%),B:657.5(1.71%),P:1e-1635\n",
      "\t\tRound 2: -3764.75 CCATTCCA T:2746.0(92.05%),B:657.5(1.71%),P:1e-1635\n",
      "\t\t=Final=: -1103.16 CCATTCCA T:428.0(39.45%),B:546.4(1.43%),P:1e-479\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: ATTCGATT... (-1771.736)\n",
      "\t\tRound 1: -2124.05 ATTCGATT T:1420.0(73.00%),B:1214.9(3.13%),P:1e-922\n",
      "\t\tRound 2: -2124.05 ATTCGATT T:1420.0(73.00%),B:1214.9(3.13%),P:1e-922\n",
      "\t\t=Final=: -892.22 ATTCGATT T:240.0(22.12%),B:87.9(0.23%),P:1e-387\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GAATCGAA... (-1576.147)\n",
      "\t\tRound 1: -1744.79 GAATCGAA T:1918.0(82.94%),B:3368.5(8.45%),P:1e-757\n",
      "\t\tRound 2: -1744.79 GAATCGAA T:1918.0(82.94%),B:3368.5(8.45%),P:1e-757\n",
      "\t\t=Final=: -763.67 GAATCGAA T:401.0(36.96%),B:1010.4(2.65%),P:1e-331\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: TCCATTCC... (-811.834)\n",
      "\t\tRound 1: -1116.95 TCCATTCC T:1311.0(70.15%),B:4090.6(10.16%),P:1e-485\n",
      "\t\tRound 2: -1116.95 TCCATTCC T:1311.0(70.15%),B:4090.6(10.16%),P:1e-485\n",
      "\t\t=Final=: -516.87 TCCATTCC T:362.0(33.36%),B:1484.3(3.89%),P:1e-224\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: ATTCCATT... (-530.104)\n",
      "\t\tRound 1: -773.55 ATTCCATT T:1193.0(66.71%),B:5902.6(14.33%),P:1e-335\n",
      "\t\tRound 2: -773.55 ATTCCATT T:1193.0(66.71%),B:5902.6(14.33%),P:1e-335\n",
      "\t\t=Final=: -387.79 ATTCCATT T:354.0(32.63%),B:2090.1(5.48%),P:1e-168\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: ATTCGAGT... (-268.377)\n",
      "\t\tRound 1: -399.24 ATTCGATT T:265.0(21.68%),B:684.7(1.78%),P:1e-173\n",
      "\t\tRound 2: -399.24 ATTCGATT T:265.0(21.68%),B:684.7(1.78%),P:1e-173\n",
      "\t\t=Final=: -346.65 ATTCGATT T:214.0(19.72%),B:665.8(1.74%),P:1e-150\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: TTCCGTTT... (-226.134)\n",
      "\t\tRound 1: -340.83 TTCCRTTY T:1248.0(68.36%),B:13607.7(29.99%),P:1e-148\n",
      "\t\tRound 2: -391.27 TTCCRTTY T:1754.0(80.16%),B:18718.1(38.76%),P:1e-169\n",
      "\t\tRound 3: -409.66 TTCCATTY T:1491.0(74.71%),B:14948.4(32.41%),P:1e-177\n",
      "\t\tRound 4: -422.44 TTHCRTTC T:485.0(36.06%),B:2470.0(6.27%),P:1e-183\n",
      "\t\tRound 5: -422.44 TTHCRTTC T:485.0(36.06%),B:2470.0(6.27%),P:1e-183\n",
      "\t\t=Final=: -410.61 TTHCRTTC T:382.0(35.21%),B:2340.6(6.13%),P:1e-178\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: GTTTCATT... (-143.157)\n",
      "\t\tRound 1: -239.52 GTTTCATT T:237.0(19.63%),B:1150.5(2.97%),P:1e-104\n",
      "\t\tRound 2: -239.52 GTTTCATT T:237.0(19.63%),B:1150.5(2.97%),P:1e-104\n",
      "\t\t=Final=: -226.53 GTTTCATT T:206.0(18.99%),B:1136.8(2.98%),P:1e-98\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: GAGTCCGT... (-82.672)\n",
      "\t\tRound 1: -82.67 GAGTCCGT T:35.0(3.18%),B:45.6(0.12%),P:1e-35\n",
      "\t\tRound 2: -82.67 GAGTCCGT T:35.0(3.18%),B:45.6(0.12%),P:1e-35\n",
      "\t\t=Final=: -85.27 GAGTCCGT T:35.0(3.23%),B:45.6(0.12%),P:1e-37\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: CCTTATAA... (-75.184)\n",
      "\t\tRound 1: -91.76 CCTTTTAA T:225.0(18.74%),B:2614.4(6.62%),P:1e-39\n",
      "\t\tRound 2: -91.76 CCTTTTAA T:225.0(18.74%),B:2614.4(6.62%),P:1e-39\n",
      "\t\t=Final=: -63.42 CCTTTTAA T:113.0(10.41%),B:1185.6(3.11%),P:1e-27\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: AAATCGAA... (-75.147)\n",
      "\t\tRound 1: -120.37 AAATCGAA T:22.0(2.01%),B:0.0(0.00%),P:1e-52\n",
      "\t\tRound 2: -120.37 AAATCGAA T:22.0(2.01%),B:0.0(0.00%),P:1e-52\n",
      "\t\t=Final=: -127.04 AAATCGAA T:22.0(2.03%),B:0.0(0.00%),P:1e-55\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: GTCAGCCC... (-70.878)\n",
      "\t\tRound 1: -70.88 GTCAGCCC T:44.0(3.98%),B:127.9(0.33%),P:1e-30\n",
      "\t\tRound 2: -70.88 GTCAGCCC T:44.0(3.98%),B:127.9(0.33%),P:1e-30\n",
      "\t\t=Final=: -73.42 GTCAGCCC T:44.0(4.06%),B:126.6(0.33%),P:1e-31\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: TCCAATCC... (-70.350)\n",
      "\t\tRound 1: -126.44 TCCAATCC T:178.0(15.14%),B:1345.1(3.46%),P:1e-54\n",
      "\t\tRound 2: -126.44 TCCAATCC T:178.0(15.14%),B:1345.1(3.46%),P:1e-54\n",
      "\t\t=Final=: -95.19 TCCAATCC T:86.0(7.93%),B:456.3(1.20%),P:1e-41\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TTTGATTC... (-69.974)\n",
      "\t\tRound 1: -100.43 TTTGATTC T:167.0(14.27%),B:1468.1(3.77%),P:1e-43\n",
      "\t\tRound 2: -100.43 TTTGATTC T:167.0(14.27%),B:1468.1(3.77%),P:1e-43\n",
      "\t\t=Final=: -98.43 TTTGATTC T:152.0(14.01%),B:1429.3(3.75%),P:1e-42\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: TAGGTTAG... (-65.058)\n",
      "\t\tRound 1: -65.06 TAGGTTAG T:33.0(3.00%),B:64.4(0.17%),P:1e-28\n",
      "\t\tRound 2: -65.06 TAGGTTAG T:33.0(3.00%),B:64.4(0.17%),P:1e-28\n",
      "\t\t=Final=: -3.28 TAGGTTAG T:5.0(0.46%),B:64.0(0.17%),P:1e-1\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: GAATGTAG... (-64.752)\n",
      "\t\tRound 1: -106.32 GAATGTAA T:158.0(13.56%),B:1274.3(3.28%),P:1e-46\n",
      "\t\tRound 2: -106.32 GAATGTAA T:158.0(13.56%),B:1274.3(3.28%),P:1e-46\n",
      "\t\t=Final=: -111.75 GAATGTAA T:151.0(13.92%),B:1260.2(3.30%),P:1e-48\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: ACTGGCAG... (-59.242)\n",
      "\t\tRound 1: -59.24 ACTGGCAG T:49.0(4.42%),B:212.3(0.55%),P:1e-25\n",
      "\t\tRound 2: -59.24 ACTGGCAG T:49.0(4.42%),B:212.3(0.55%),P:1e-25\n",
      "\t\t=Final=: -63.50 ACTGGCAG T:49.0(4.52%),B:211.9(0.56%),P:1e-27\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: GGTGTAGT... (-55.637)\n",
      "\t\tRound 1: -59.71 GGTGTWGW T:181.0(15.37%),B:2429.3(6.17%),P:1e-25\n",
      "\t\tRound 2: -77.22 GGWRTMGW T:119.0(10.39%),B:1004.2(2.60%),P:1e-33\n",
      "\t\tRound 3: -88.64 GGAMTCGA T:356.0(27.98%),B:5285.9(12.93%),P:1e-38\n",
      "\t\tRound 4: -93.96 GKRSWCGA T:64.0(5.73%),B:209.2(0.55%),P:1e-40\n",
      "\t\tRound 5: -93.96 GKRSWCGA T:64.0(5.73%),B:209.2(0.55%),P:1e-40\n",
      "\t\t=Final=: -97.77 GKRSWCGA T:63.0(5.81%),B:203.0(0.53%),P:1e-42\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: GTGCTCTG... (-54.327)\n",
      "\t\tRound 1: -66.22 KTGCTCTG T:126.0(10.97%),B:1279.8(3.30%),P:1e-28\n",
      "\t\tRound 2: -66.22 KTGCTCTG T:126.0(10.97%),B:1279.8(3.30%),P:1e-28\n",
      "\t\t=Final=: -36.31 KTGCTCTG T:93.0(8.57%),B:1250.7(3.28%),P:1e-15\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: GTTTGGTG... (-54.612)\n",
      "\t\tRound 1: -54.61 GTTTGGTG T:43.0(3.89%),B:182.6(0.48%),P:1e-23\n",
      "\t\tRound 2: -54.61 GTTTGGTG T:43.0(3.89%),B:182.6(0.48%),P:1e-23\n",
      "\t\t=Final=: -4.98 GTTTGGTG T:12.0(1.11%),B:182.1(0.48%),P:1e-2\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: AGGAATGG... (-52.675)\n",
      "\t\tRound 1: -69.00 TGGAATGG T:372.0(29.04%),B:6324.7(15.27%),P:1e-29\n",
      "\t\tRound 2: -80.98 TGKAATGG T:227.0(18.89%),B:2858.6(7.21%),P:1e-35\n",
      "\t\tRound 3: -80.98 TGKAATGG T:227.0(18.89%),B:2858.6(7.21%),P:1e-35\n",
      "\t\t=Final=: -59.40 TGKAATGG T:177.0(16.31%),B:2611.1(6.84%),P:1e-25\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: TTTCGCTT... (-51.111)\n",
      "\t\tRound 1: -103.94 TTTCGSTT T:75.0(6.68%),B:262.4(0.68%),P:1e-45\n",
      "\t\tRound 2: -103.94 TTTCGSTT T:75.0(6.68%),B:262.4(0.68%),P:1e-45\n",
      "\t\t=Final=: -104.94 TTTCGSTT T:72.0(6.64%),B:257.7(0.68%),P:1e-45\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: GCATACTA... (-49.397)\n",
      "\t\tRound 1: -96.73 GCATACTA T:81.0(7.20%),B:357.0(0.93%),P:1e-42\n",
      "\t\tRound 2: -96.73 GCATACTA T:81.0(7.20%),B:357.0(0.93%),P:1e-42\n",
      "\t\t=Final=: -61.28 GCATACTA T:60.0(5.53%),B:354.0(0.93%),P:1e-26\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: CTATGCGA... (-49.075)\n",
      "\t\tRound 1: -49.07 CTATGCGA T:19.0(1.74%),B:19.4(0.05%),P:1e-21\n",
      "\t\tRound 2: -49.07 CTATGCGA T:19.0(1.74%),B:19.4(0.05%),P:1e-21\n",
      "\t\t=Final=: -51.70 CTATGCGA T:19.0(1.75%),B:19.4(0.05%),P:1e-22\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: TGCACCTA... (-47.702)\n",
      "\t\tRound 1: -63.19 TGCACCTA T:53.0(4.77%),B:235.1(0.61%),P:1e-27\n",
      "\t\tRound 2: -63.19 TGCACCTA T:53.0(4.77%),B:235.1(0.61%),P:1e-27\n",
      "\t\t=Final=: -49.91 TGCACCTA T:30.0(2.76%),B:88.6(0.23%),P:1e-21\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: STAT1_6h_IFNa_homer/homerMotifs.motifs8\n",
      "\n",
      "\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 496643\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 496643 of 551887 max\n",
      "\tTree  : 1023352 of 2759435 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 496643 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t81.68% skipped, 18.32% checked (90994 of 496643), of those checked:\n",
      "\t\t81.68% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 496643 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t81.68% skipped, 18.32% checked (90994 of 496643), of those checked:\n",
      "\t\t0.00% not in target, 14.45% increased p-value, 9.22% high p-value\n",
      "\n",
      "\tScreening oligos 496643 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t96.72% skipped, 3.28% checked (16277 of 496643), of those checked:\n",
      "\t\t0.00% not in target, 1.61% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: ATTCCATTCC... (-4925.159)\n",
      "\t\tRound 1: -5115.56 ATTCCATTCG T:2786.0(92.34%),B:173.2(0.45%),P:1e-2221\n",
      "\t\tRound 2: -5115.56 ATTCCATTCG T:2786.0(92.34%),B:173.2(0.45%),P:1e-2221\n",
      "\t\t=Final=: -1878.62 ATTCCATTCG T:424.0(39.08%),B:84.0(0.22%),P:1e-815\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: CCATTCCATT... (-2891.108)\n",
      "\t\tRound 1: -3129.01 CCATTCCATT T:2140.0(86.10%),B:854.4(2.21%),P:1e-1358\n",
      "\t\tRound 2: -3129.01 CCATTCCATT T:2140.0(86.10%),B:854.4(2.21%),P:1e-1358\n",
      "\t\t=Final=: -1339.24 CCATTCCATT T:304.0(28.02%),B:57.8(0.15%),P:1e-581\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: TCCATTCCAT... (-2308.913)\n",
      "\t\tRound 1: -2468.28 TCCATTCCAT T:1857.0(81.96%),B:1376.1(3.54%),P:1e-1071\n",
      "\t\tRound 2: -2468.28 TCCATTCCAT T:1857.0(81.96%),B:1376.1(3.54%),P:1e-1071\n",
      "\t\t=Final=: -950.15 TCCATTCCAT T:414.0(38.16%),B:700.2(1.83%),P:1e-412\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: GGAATCGAAT... (-937.639)\n",
      "\t\tRound 1: -1294.19 GGAATGGAAT T:1533.0(75.67%),B:4161.3(10.33%),P:1e-562\n",
      "\t\tRound 2: -1353.26 GGAATGGAAT T:912.0(56.87%),B:1346.2(3.47%),P:1e-587\n",
      "\t\tRound 3: -1353.26 GGAATGGAAT T:912.0(56.87%),B:1346.2(3.47%),P:1e-587\n",
      "\t\t=Final=: -671.94 GGAATGGAAT T:353.0(32.53%),B:867.6(2.27%),P:1e-291\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: AATCGAATCG... (-487.373)\n",
      "\t\tRound 1: -773.46 AATSGAATSG T:1059.0(62.34%),B:4853.6(11.94%),P:1e-335\n",
      "\t\tRound 2: -773.46 AATSGAATSG T:1059.0(62.34%),B:4853.6(11.94%),P:1e-335\n",
      "\t\t=Final=: -435.20 AATSGAATSG T:344.0(31.71%),B:1671.9(4.38%),P:1e-189\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: GAAACCGAAA... (-358.131)\n",
      "\t\tRound 1: -536.80 GAAAVYGAAA T:390.0(30.21%),B:1061.3(2.74%),P:1e-233\n",
      "\t\tRound 2: -582.52 GAAASTGAAA T:388.0(30.08%),B:904.5(2.34%),P:1e-252\n",
      "\t\tRound 3: -582.52 GAAASTGAAA T:388.0(30.08%),B:904.5(2.34%),P:1e-252\n",
      "\t\t=Final=: -605.15 GAAASTGAAA T:331.0(30.51%),B:870.3(2.28%),P:1e-262\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: AAAATCGACT... (-195.980)\n",
      "\t\tRound 1: -195.98 AAAATCGACT T:33.0(3.00%),B:1.2(0.00%),P:1e-85\n",
      "\t\tRound 2: -195.98 AAAATCGACT T:33.0(3.00%),B:1.2(0.00%),P:1e-85\n",
      "\t\t=Final=: -203.06 AAAATCGACT T:33.0(3.04%),B:1.2(0.00%),P:1e-88\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: AGTTCATTAC... (-186.347)\n",
      "\t\tRound 1: -215.76 ATTNCATTAC T:273.0(22.25%),B:1751.7(4.48%),P:1e-93\n",
      "\t\tRound 2: -285.93 ATTHCATTAY T:271.0(22.11%),B:1216.6(3.14%),P:1e-124\n",
      "\t\tRound 3: -285.93 ATTHCATTAY T:271.0(22.11%),B:1216.6(3.14%),P:1e-124\n",
      "\t\t=Final=: -164.09 ATTHCATTAY T:155.0(14.29%),B:881.2(2.31%),P:1e-71\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: GGTTAGGTTA... (-181.902)\n",
      "\t\tRound 1: -181.90 GGTTAGGTTA T:31.0(2.82%),B:2.8(0.01%),P:1e-78\n",
      "\t\tRound 2: -181.90 GGTTAGGTTA T:31.0(2.82%),B:2.8(0.01%),P:1e-78\n",
      "\t\t=Final=: -19.18 GGTTAGGTTA T:5.0(0.46%),B:2.8(0.01%),P:1e-8\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: GTCAGCCCAT... (-160.139)\n",
      "\t\tRound 1: -165.55 GTCAGCCCAT T:38.0(3.44%),B:7.7(0.02%),P:1e-71\n",
      "\t\tRound 2: -165.55 GTCAGCCCAT T:38.0(3.44%),B:7.7(0.02%),P:1e-71\n",
      "\t\t=Final=: -165.16 GTCAGCCCAT T:38.0(3.50%),B:7.7(0.02%),P:1e-71\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: GGTGTAGTTG... (-147.319)\n",
      "\t\tRound 1: -160.73 GGTGTAGTTG T:40.0(3.62%),B:10.5(0.03%),P:1e-69\n",
      "\t\tRound 2: -160.73 GGTGTAGTTG T:40.0(3.62%),B:10.5(0.03%),P:1e-69\n",
      "\t\t=Final=: -33.49 GGTGTAGTTG T:7.0(0.65%),B:0.9(0.00%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: ACAGAATGTA... (-146.494)\n",
      "\t\tRound 1: -148.28 ACAGAATGTA T:41.0(3.71%),B:15.2(0.04%),P:1e-64\n",
      "\t\tRound 2: -152.09 ACGGAACGTA T:43.0(3.89%),B:17.6(0.04%),P:1e-66\n",
      "\t\tRound 3: -152.09 ACGGAACGTA T:43.0(3.89%),B:17.6(0.04%),P:1e-66\n",
      "\t\t=Final=: -154.10 ACGGAACGTA T:43.0(3.96%),B:17.6(0.05%),P:1e-66\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: TTGCATTAGA... (-144.585)\n",
      "\t\tRound 1: -144.58 TTGCATTAGA T:33.0(3.00%),B:6.8(0.02%),P:1e-62\n",
      "\t\tRound 2: -144.58 TTGCATTAGA T:33.0(3.00%),B:6.8(0.02%),P:1e-62\n",
      "\t\t=Final=: -144.07 TTGCATTAGA T:33.0(3.04%),B:6.8(0.02%),P:1e-62\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: CTGTGGTGGT... (-140.591)\n",
      "\t\tRound 1: -140.59 CTGTGGTGGT T:41.0(3.71%),B:18.6(0.05%),P:1e-61\n",
      "\t\tRound 2: -140.59 CTGTGGTGGT T:41.0(3.71%),B:18.6(0.05%),P:1e-61\n",
      "\t\t=Final=: -13.68 CTGTGGTGGT T:7.0(0.65%),B:18.6(0.05%),P:1e-5\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: GATAGAGTAG... (-139.139)\n",
      "\t\tRound 1: -232.25 RATGGAGTAG T:137.0(11.87%),B:315.2(0.82%),P:1e-100\n",
      "\t\tRound 2: -239.89 RATGGAGTAG T:91.0(8.05%),B:80.7(0.21%),P:1e-104\n",
      "\t\tRound 3: -239.89 RATGGAGTAG T:91.0(8.05%),B:80.7(0.21%),P:1e-104\n",
      "\t\t=Final=: -235.09 RATGGAGTAG T:86.0(7.93%),B:80.7(0.21%),P:1e-102\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: CGAATGGATT... (-136.166)\n",
      "\t\tRound 1: -228.31 CGAATGGAAT T:44.0(3.98%),B:4.3(0.01%),P:1e-99\n",
      "\t\tRound 2: -228.31 CGAATGGAAT T:44.0(3.98%),B:4.3(0.01%),P:1e-99\n",
      "\t\t=Final=: -221.96 CGAATGGAAT T:44.0(4.06%),B:4.3(0.01%),P:1e-96\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: CCTTATAAGG... (-133.758)\n",
      "\t\tRound 1: -133.76 CCTTATAAGG T:24.0(2.19%),B:2.6(0.01%),P:1e-58\n",
      "\t\tRound 2: -133.76 CCTTATAAGG T:24.0(2.19%),B:2.6(0.01%),P:1e-58\n",
      "\t\t=Final=: -123.91 CCTTATAAGG T:24.0(2.21%),B:2.6(0.01%),P:1e-53\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: CTCTAGTCTA... (-131.433)\n",
      "\t\tRound 1: -131.43 CTCTAGTCTA T:28.0(2.55%),B:4.5(0.01%),P:1e-57\n",
      "\t\tRound 2: -131.43 CTCTAGTCTA T:28.0(2.55%),B:4.5(0.01%),P:1e-57\n",
      "\t\t=Final=: -129.22 CTCTAGTCTA T:28.0(2.58%),B:4.5(0.01%),P:1e-56\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: CCATCCCATG... (-125.313)\n",
      "\t\tRound 1: -182.97 CCATTCCATC T:105.0(9.23%),B:240.8(0.63%),P:1e-79\n",
      "\t\tRound 2: -224.42 CCATTCCATG T:86.0(7.62%),B:77.0(0.20%),P:1e-97\n",
      "\t\tRound 3: -224.42 CCATTCCATG T:86.0(7.62%),B:77.0(0.20%),P:1e-97\n",
      "\t\t=Final=: -238.30 CCATTCCATG T:86.0(7.93%),B:77.0(0.20%),P:1e-103\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: GAGCATTCGT... (-120.370)\n",
      "\t\tRound 1: -120.37 GAGCATTCGT T:22.0(2.01%),B:1.1(0.00%),P:1e-52\n",
      "\t\tRound 2: -120.37 GAGCATTCGT T:22.0(2.01%),B:1.1(0.00%),P:1e-52\n",
      "\t\t=Final=: -127.04 GAGCATTCGT T:22.0(2.03%),B:1.1(0.00%),P:1e-55\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: GCAGTGCAAT... (-117.924)\n",
      "\t\tRound 1: -161.20 GCASTGCAMT T:58.0(5.21%),B:46.1(0.12%),P:1e-70\n",
      "\t\tRound 2: -161.20 GCASTGCAMT T:58.0(5.21%),B:46.1(0.12%),P:1e-70\n",
      "\t\t=Final=: -87.15 GCASTGCAMT T:29.0(2.67%),B:21.0(0.06%),P:1e-37\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: CCAAAGGAGT... (-107.658)\n",
      "\t\tRound 1: -110.00 CCAAAGGAGT T:34.0(3.09%),B:18.7(0.05%),P:1e-47\n",
      "\t\tRound 2: -110.00 CCAAAGGAGT T:34.0(3.09%),B:18.7(0.05%),P:1e-47\n",
      "\t\t=Final=: -112.36 CCAAAGGAGT T:34.0(3.13%),B:18.7(0.05%),P:1e-48\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: GGAGTCCGTA... (-107.173)\n",
      "\t\tRound 1: -107.17 GGAGTCCGTA T:20.0(1.83%),B:2.3(0.01%),P:1e-46\n",
      "\t\tRound 2: -107.17 GGAGTCCGTA T:20.0(1.83%),B:2.3(0.01%),P:1e-46\n",
      "\t\t=Final=: -99.91 GGAGTCCGTA T:20.0(1.84%),B:2.3(0.01%),P:1e-43\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: ACTGGCAGGG... (-105.642)\n",
      "\t\tRound 1: -174.62 ACTGGCARGV T:94.0(8.30%),B:192.4(0.50%),P:1e-75\n",
      "\t\tRound 2: -174.62 ACTGGCARGV T:94.0(8.30%),B:192.4(0.50%),P:1e-75\n",
      "\t\t=Final=: -165.98 ACTGGCARGV T:87.0(8.02%),B:191.2(0.50%),P:1e-72\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: CTGTTTGAGT... (-103.728)\n",
      "\t\tRound 1: -105.08 CTGTTTGAGT T:28.0(2.55%),B:10.0(0.02%),P:1e-45\n",
      "\t\tRound 2: -105.08 CTGTTTGAGT T:28.0(2.55%),B:10.0(0.02%),P:1e-45\n",
      "\t\t=Final=: -106.65 CTGTTTGAGT T:28.0(2.58%),B:10.0(0.03%),P:1e-46\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: STAT1_6h_IFNa_homer/homerMotifs.motifs10\n",
      "\n",
      "\n",
      "\t-blen automatically set to 2\n",
      "\tScanning input files...\n",
      "\tParsing sequences...\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\tTotal number of Oligos: 3797474\n",
      "\tAutoadjustment for sequence coverage in background: 1.00x\n",
      "\n",
      "\tOligos: 3797474 of 8830118 max\n",
      "\tTree  : 8634424 of 44150590 max\n",
      "\tOptimizing memory usage...\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tGlobal Optimization Phase: Looking for enriched oligos with up to 2 mismatches...\n",
      "\n",
      "\tScreening oligos 3797474 (allowing 0 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t97.03% skipped, 2.97% checked (112657 of 3797474), of those checked:\n",
      "\t\t97.03% not in target, 0.00% increased p-value, 0.00% high p-value\n",
      "\n",
      "\tScreening oligos 3797474 (allowing 1 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t97.03% skipped, 2.97% checked (112657 of 3797474), of those checked:\n",
      "\t\t0.00% not in target, 2.54% increased p-value, 0.07% high p-value\n",
      "\n",
      "\tScreening oligos 3797474 (allowing 2 mismatches):\n",
      "\t|0%                                   50%                                  100%|\n",
      "\t================================================================================\n",
      "\t\t99.58% skipped, 0.42% checked (16123 of 3797474), of those checked:\n",
      "\t\t0.00% not in target, 0.18% increased p-value, 0.00% high p-value\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\n",
      "\tLocal Optimization Phase:\n",
      "\t1 of 25 Initial Sequence: TCGAATGGAATG... (-5546.935)\n",
      "\t\tRound 1: -5709.72 TGGAATGGAATG T:2965.0(93.50%),B:107.3(0.28%),P:1e-2479\n",
      "\t\tRound 2: -5820.75 TGGAATGGAATG T:2437.0(89.43%),B:66.8(0.17%),P:1e-2527\n",
      "\t\tRound 3: -5820.75 TGGAATGGAATG T:2437.0(89.43%),B:66.8(0.17%),P:1e-2527\n",
      "\t\t=Final=: -2204.44 TGGAATGGAATG T:395.0(36.41%),B:25.0(0.07%),P:1e-957\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t2 of 25 Initial Sequence: TTCCATTCCATT... (-4111.261)\n",
      "\t\tRound 1: -4407.59 TTCCATTCCATT T:2524.0(90.24%),B:300.4(0.78%),P:1e-1914\n",
      "\t\tRound 2: -4492.40 TTCCATTCCATT T:3018.0(93.81%),B:362.4(0.94%),P:1e-1951\n",
      "\t\tRound 3: -4492.40 TTCCATTCCATT T:3018.0(93.81%),B:362.4(0.94%),P:1e-1951\n",
      "\t\t=Final=: -1795.05 TTCCATTCCATT T:370.0(34.10%),B:46.9(0.12%),P:1e-779\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t3 of 25 Initial Sequence: GAATGGAATCGA... (-3233.917)\n",
      "\t\tRound 1: -3637.53 GAATGGAATGGA T:1937.0(83.24%),B:400.9(1.04%),P:1e-1579\n",
      "\t\tRound 2: -3637.53 GAATGGAATGGA T:1937.0(83.24%),B:400.9(1.04%),P:1e-1579\n",
      "\t\t=Final=: -1586.32 GAATGGAATGGA T:403.0(37.14%),B:129.0(0.34%),P:1e-688\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t4 of 25 Initial Sequence: TTCGATTCCGTT... (-1546.802)\n",
      "\t\tRound 1: -1935.61 TTCCATTCCATT T:1830.0(81.50%),B:2499.4(6.34%),P:1e-840\n",
      "\t\tRound 2: -2049.32 TTCCATTCCATT T:1340.0(70.93%),B:1172.2(3.02%),P:1e-890\n",
      "\t\tRound 3: -2082.97 TTCCATTCCATT T:1530.0(75.60%),B:1496.1(3.84%),P:1e-904\n",
      "\t\tRound 4: -2094.02 TTCCATTCCATT T:1356.0(71.36%),B:1137.2(2.94%),P:1e-909\n",
      "\t\tRound 5: -2094.02 TTCCATTCCATT T:1356.0(71.36%),B:1137.2(2.94%),P:1e-909\n",
      "\t\t=Final=: -882.96 TTCCATTCCATT T:348.0(32.07%),B:442.8(1.16%),P:1e-383\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t5 of 25 Initial Sequence: CCATTCCATTCG... (-864.916)\n",
      "\t\tRound 1: -1094.80 CCATTCCATTCG T:354.0(27.85%),B:126.0(0.33%),P:1e-475\n",
      "\t\tRound 2: -1094.80 CCATTCCATTCG T:354.0(27.85%),B:126.0(0.33%),P:1e-475\n",
      "\t\t=Final=: -815.01 CCATTCCATTCG T:242.0(22.30%),B:125.8(0.33%),P:1e-353\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t6 of 25 Initial Sequence: GAAACCGAAACT... (-610.780)\n",
      "\t\tRound 1: -940.70 GAAAMYGAAACT T:370.0(28.91%),B:243.7(0.63%),P:1e-408\n",
      "\t\tRound 2: -982.01 GAAAAYGAAACT T:521.0(38.15%),B:647.4(1.68%),P:1e-426\n",
      "\t\tRound 3: -982.01 GAAAAYGAAACT T:521.0(38.15%),B:647.4(1.68%),P:1e-426\n",
      "\t\t=Final=: -1031.87 GAAAAYGAAACT T:399.0(36.77%),B:496.9(1.30%),P:1e-448\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t7 of 25 Initial Sequence: ATAACGAAACTT... (-283.184)\n",
      "\t\tRound 1: -327.47 ATAACGAAACTT T:57.0(5.12%),B:3.9(0.01%),P:1e-142\n",
      "\t\tRound 2: -327.47 ATAACGAAACTT T:57.0(5.12%),B:3.9(0.01%),P:1e-142\n",
      "\t\t=Final=: -247.24 ATAACGAAACTT T:46.0(4.24%),B:3.9(0.01%),P:1e-107\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t8 of 25 Initial Sequence: TGGAATGGACTG... (-278.375)\n",
      "\t\tRound 1: -544.99 TGGAATGGAATG T:894.0(56.15%),B:5572.8(13.58%),P:1e-236\n",
      "\t\tRound 2: -599.05 TGGAATGGAATG T:520.0(38.09%),B:1736.5(4.45%),P:1e-260\n",
      "\t\tRound 3: -611.80 TGGAATGGAATG T:406.0(31.23%),B:920.3(2.38%),P:1e-265\n",
      "\t\tRound 4: -611.80 TGGAATGGAATG T:406.0(31.23%),B:920.3(2.38%),P:1e-265\n",
      "\t\t=Final=: -310.49 TGGAATGGAATG T:148.0(13.64%),B:270.1(0.71%),P:1e-134\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t9 of 25 Initial Sequence: CTCTGTGGTGGT... (-246.206)\n",
      "\t\tRound 1: -246.21 CTCTGTGGTGGT T:40.0(3.62%),B:0.0(0.00%),P:1e-106\n",
      "\t\tRound 2: -246.21 CTCTGTGGTGGT T:40.0(3.62%),B:0.0(0.00%),P:1e-106\n",
      "\t\t=Final=: -33.49 CTCTGTGGTGGT T:7.0(0.65%),B:0.0(0.00%),P:1e-14\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t10 of 25 Initial Sequence: AATTAAATGGAC... (-224.509)\n",
      "\t\tRound 1: -353.01 AATYAAATGGAM T:124.0(10.80%),B:83.8(0.22%),P:1e-153\n",
      "\t\tRound 2: -365.42 AATCAAATGGAA T:123.0(10.72%),B:72.4(0.19%),P:1e-158\n",
      "\t\tRound 3: -385.24 AATCAAATGGAA T:108.0(9.48%),B:34.4(0.09%),P:1e-167\n",
      "\t\tRound 4: -408.51 AATAAAATSGAM T:276.0(22.47%),B:724.0(1.88%),P:1e-177\n",
      "\t\tRound 5: -427.14 AATARAATGGAA T:195.0(16.46%),B:241.6(0.63%),P:1e-185\n",
      "\t\tRound 6: -427.14 AATARAATGGAA T:195.0(16.46%),B:241.6(0.63%),P:1e-185\n",
      "\t\t=Final=: -376.34 AATARAATGGAA T:163.0(15.02%),B:240.6(0.63%),P:1e-163\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t11 of 25 Initial Sequence: ATGCCATCCCAT... (-210.184)\n",
      "\t\tRound 1: -224.51 ATGCCATCCCAT T:37.0(3.35%),B:2.6(0.01%),P:1e-97\n",
      "\t\tRound 2: -245.85 ATSCCATTCCAT T:56.0(5.03%),B:9.1(0.02%),P:1e-106\n",
      "\t\tRound 3: -245.85 ATSCCATTCCAT T:56.0(5.03%),B:9.1(0.02%),P:1e-106\n",
      "\t\t=Final=: -250.37 ATSCCATTCCAT T:56.0(5.16%),B:9.1(0.02%),P:1e-108\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t12 of 25 Initial Sequence: AGAATGTAGAGG... (-203.067)\n",
      "\t\tRound 1: -217.33 AGAATGTAGAGG T:36.0(3.27%),B:2.3(0.01%),P:1e-94\n",
      "\t\tRound 2: -224.51 AGAATGTAGAGG T:37.0(3.35%),B:2.7(0.01%),P:1e-97\n",
      "\t\tRound 3: -224.51 AGAATGTAGAGG T:37.0(3.35%),B:2.7(0.01%),P:1e-97\n",
      "\t\t=Final=: -210.18 AGAATGTAGAGG T:34.0(3.13%),B:1.6(0.00%),P:1e-91\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t13 of 25 Initial Sequence: ACACCATTCCAA... (-167.960)\n",
      "\t\tRound 1: -170.21 ACACCATTCCAA T:54.0(4.86%),B:29.1(0.08%),P:1e-73\n",
      "\t\tRound 2: -206.72 ABTCCGTTCCAT T:73.0(6.51%),B:53.3(0.14%),P:1e-89\n",
      "\t\tRound 3: -242.45 ABWCCHTTCCAT T:74.0(6.60%),B:34.4(0.09%),P:1e-105\n",
      "\t\tRound 4: -242.45 ABWCCHTTCCAT T:74.0(6.60%),B:34.4(0.09%),P:1e-105\n",
      "\t\t=Final=: -253.52 ABWCCHTTCCAT T:74.0(6.82%),B:34.4(0.09%),P:1e-110\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t14 of 25 Initial Sequence: TATCTCCACACT... (-167.960)\n",
      "\t\tRound 1: -174.91 TATCTCCACACT T:30.0(2.73%),B:1.9(0.00%),P:1e-75\n",
      "\t\tRound 2: -188.92 TATCTCAACACD T:32.0(2.91%),B:2.6(0.01%),P:1e-82\n",
      "\t\tRound 3: -188.92 TATCTCAACACD T:32.0(2.91%),B:2.6(0.01%),P:1e-82\n",
      "\t\t=Final=: -188.92 TATCTCAACACD T:31.0(2.86%),B:1.2(0.00%),P:1e-82\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t15 of 25 Initial Sequence: CTATACCATTGC... (-161.042)\n",
      "\t\tRound 1: -176.73 CTATACCATTGC T:44.0(3.98%),B:11.4(0.03%),P:1e-76\n",
      "\t\tRound 2: -224.51 CYRTASCATTSC T:37.0(3.35%),B:2.0(0.00%),P:1e-97\n",
      "\t\tRound 3: -224.51 CYRTASCATTSC T:37.0(3.35%),B:2.0(0.00%),P:1e-97\n",
      "\t\t=Final=: -231.71 CYRTASCATTSC T:37.0(3.41%),B:2.0(0.01%),P:1e-100\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t16 of 25 Initial Sequence: TTAGGTTAGGTT... (-161.042)\n",
      "\t\tRound 1: -195.98 TTAGGTTAGGTT T:33.0(3.00%),B:2.0(0.01%),P:1e-85\n",
      "\t\tRound 2: -195.98 TTAGGTTAGGTT T:33.0(3.00%),B:2.0(0.01%),P:1e-85\n",
      "\t\t=Final=: -23.84 TTAGGTTAGGTT T:6.0(0.55%),B:2.0(0.01%),P:1e-10\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t17 of 25 Initial Sequence: TACTCTATCTGA... (-154.839)\n",
      "\t\tRound 1: -167.46 TACTCTATCTGA T:32.0(2.91%),B:3.8(0.01%),P:1e-72\n",
      "\t\tRound 2: -167.46 TACTCTATCTGA T:32.0(2.91%),B:3.8(0.01%),P:1e-72\n",
      "\t\t=Final=: -160.87 TACTCTATCTGA T:32.0(2.95%),B:3.8(0.01%),P:1e-69\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t18 of 25 Initial Sequence: CGATTGCATTGC... (-154.161)\n",
      "\t\tRound 1: -12.06 CGATTGCATTGC T:14.0(1.28%),B:97.3(0.25%),P:1e-5\n",
      "\t\tRound 2: -87.78 CCATTACATTCC T:17.0(1.56%),B:0.0(0.00%),P:1e-38\n",
      "\t\tRound 3: -113.75 YCTTTHBATTCC T:21.0(1.92%),B:1.8(0.00%),P:1e-49\n",
      "\t\tRound 4: -117.69 CCWTTMCTTTCG T:28.0(2.55%),B:6.3(0.02%),P:1e-51\n",
      "\t\tRound 5: -150.63 CCWTTAYATTCC T:38.0(3.44%),B:10.9(0.03%),P:1e-65\n",
      "\t\tRound 6: -150.63 CCWTTAYATTCC T:38.0(3.44%),B:10.9(0.03%),P:1e-65\n",
      "\t\t=Final=: -151.68 CCWTTAYATTCC T:38.0(3.50%),B:10.9(0.03%),P:1e-65\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t19 of 25 Initial Sequence: GGTCACGATAAG... (-154.161)\n",
      "\t\tRound 1: -224.51 GGTCACGATAAG T:37.0(3.35%),B:2.2(0.01%),P:1e-97\n",
      "\t\tRound 2: -224.51 GGTCACGATAAG T:37.0(3.35%),B:2.2(0.01%),P:1e-97\n",
      "\t\t=Final=: -206.09 GGTCACGATAAG T:37.0(3.41%),B:2.2(0.01%),P:1e-89\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t20 of 25 Initial Sequence: GCTCATGCACCT... (-149.572)\n",
      "\t\tRound 1: -198.12 GCTCATGCACCT T:41.0(3.71%),B:5.9(0.01%),P:1e-86\n",
      "\t\tRound 2: -210.18 GCTCATGCACCT T:35.0(3.18%),B:2.8(0.01%),P:1e-91\n",
      "\t\tRound 3: -210.18 GCTCATGCACCT T:35.0(3.18%),B:2.8(0.01%),P:1e-91\n",
      "\t\t=Final=: -193.09 GCTCATGCACCT T:35.0(3.23%),B:2.8(0.01%),P:1e-83\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t21 of 25 Initial Sequence: GTTTGGTGTAGT... (-148.277)\n",
      "\t\tRound 1: -268.14 GTTTGGTGTAGT T:43.0(3.89%),B:2.5(0.01%),P:1e-116\n",
      "\t\tRound 2: -268.14 GTTTGGTGTAGT T:43.0(3.89%),B:2.5(0.01%),P:1e-116\n",
      "\t\t=Final=: -49.14 GTTTGGTGTAGT T:11.0(1.01%),B:2.9(0.01%),P:1e-21\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t22 of 25 Initial Sequence: AGCATTCCATTG... (-147.319)\n",
      "\t\tRound 1: -164.84 ADCATTCCATTC T:50.0(4.51%),B:23.0(0.06%),P:1e-71\n",
      "\t\tRound 2: -215.97 ANCATTCCATTC T:66.0(5.90%),B:32.5(0.08%),P:1e-93\n",
      "\t\tRound 3: -215.97 ANCATTCCATTC T:66.0(5.90%),B:32.5(0.08%),P:1e-93\n",
      "\t\t=Final=: -223.47 ANCATTCCATTC T:68.0(6.27%),B:36.8(0.10%),P:1e-97\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t23 of 25 Initial Sequence: CACCTTATAAGG... (-143.576)\n",
      "\t\tRound 1: -297.73 MMCCTTATAAGG T:47.0(4.24%),B:2.3(0.01%),P:1e-129\n",
      "\t\tRound 2: -320.15 MCCCTTATAAGG T:51.0(4.59%),B:2.9(0.01%),P:1e-139\n",
      "\t\tRound 3: -320.15 MCCCTTATAAGG T:51.0(4.59%),B:2.9(0.01%),P:1e-139\n",
      "\t\t=Final=: -173.82 MCCCTTATAAGG T:32.0(2.95%),B:2.9(0.01%),P:1e-75\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t24 of 25 Initial Sequence: CTTTCAATTTGA... (-127.041)\n",
      "\t\tRound 1: -147.32 CTTTCAATTTGA T:26.0(2.37%),B:1.4(0.00%),P:1e-63\n",
      "\t\tRound 2: -147.32 CTTTCAATTTGA T:26.0(2.37%),B:1.4(0.00%),P:1e-63\n",
      "\t\t=Final=: -154.16 CTTTCAATTTGA T:26.0(2.40%),B:1.4(0.00%),P:1e-66\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\t25 of 25 Initial Sequence: CTTTTGATTCCC... (-127.041)\n",
      "\t\tRound 1: -165.55 CTTTTGAYTCCM T:38.0(3.44%),B:7.7(0.02%),P:1e-71\n",
      "\t\tRound 2: -174.91 YKTTCGATTCCC T:30.0(2.73%),B:0.0(0.00%),P:1e-75\n",
      "\t\tRound 3: -174.91 YKTTCGATTCCC T:30.0(2.73%),B:0.0(0.00%),P:1e-75\n",
      "\t\t=Final=: -181.90 YKTTCGATTCCC T:30.0(2.76%),B:0.0(0.00%),P:1e-78\n",
      "\t\tPerforming exhaustive masking of motif...\n",
      "\t\tReprioritizing potential motifs...\n",
      "\n",
      "\tFinalizing Enrichment Statistics (new in v3.4)\n",
      "\tReading input files...\n",
      "\t49999 total sequences read\n",
      "\tCache length = 11180\n",
      "\tUsing binomial scoring\n",
      "\tChecking enrichment of 25 motif(s)\n",
      "\t|0%                                    50%                                  100%|\n",
      "\t=================================================================================\n",
      "\tOutput in file: STAT1_6h_IFNa_homer/homerMotifs.motifs12\n",
      "\n",
      "\t(Motifs in homer2 format)\n",
      "\tDetermining similar motifs... 75 reduced to 25 motifs\n",
      "\tOutputing HTML and sequence logos for motif comparison...\n",
      "\tChecking de novo motifs against known motifs...\n",
      "\tFormatting HTML page...\n",
      "\t\t1 of 25 (1e-957) similar to TEAD3(TEA)/HepG2-TEAD3-ChIP-Seq(Encode)/Homer(0.706)\n",
      "\t\t2 of 25 (1e-448) similar to IRF3(IRF)/BMDM-Irf3-ChIP-Seq(GSE67343)/Homer(0.984)\n",
      "\t\t3 of 25 (1e-218) similar to PB0098.1_Zfp410_1/Jaspar(0.628)\n",
      "\t\t4 of 25 (1e-204) similar to ZBTB12(Zf)/HEK293-ZBTB12.GFP-ChIP-Seq(GSE58341)/Homer(0.588)\n",
      "\t\t5 of 25 (1e-151) similar to Arid3a/MA0151.1/Jaspar(0.777)\n",
      "\t\t6 of 25 (1e-131) similar to PB0195.1_Zbtb3_2/Jaspar(0.786)\n",
      "\t\t7 of 25 (1e-91) similar to ZNF135/MA1587.1/Jaspar(0.567)\n",
      "\t\t8 of 25 (1e-89) similar to PAX5/MA0014.3/Jaspar(0.629)\n",
      "\t\t9 of 25 (1e-88) similar to ONECUT2/MA0756.1/Jaspar(0.676)\n",
      "\t\t10 of 25 (1e-83) similar to NRF(NRF)/Promoter/Homer(0.666)\n",
      "\t\t11 of 25 (1e-82) similar to Tbx20(T-box)/Heart-Tbx20-ChIP-Seq(GSE29636)/Homer(0.597)\n",
      "\t\t12 of 25 (1e-75) similar to CArG(MADS)/PUER-Srf-ChIP-Seq(Sullivan_et_al.)/Homer(0.805)\n",
      "\t\t13 of 25 (1e-71) similar to SREBF2/MA0596.1/Jaspar(0.796)\n",
      "\t\t14 of 25 (1e-63) similar to TEAD2/MA1121.1/Jaspar(0.647)\n",
      "\t\t15 of 25 (1e-62) similar to Atf4(bZIP)/MEF-Atf4-ChIP-Seq(GSE35681)/Homer(0.723)\n",
      "\t\t16 of 25 (1e-56) similar to GATA1/MA0035.4/Jaspar(0.677)\n",
      "\t\t17 of 25 (1e-55) similar to PB0099.1_Zfp691_1/Jaspar(0.607)\n",
      "\t\t18 of 25 (1e-48) similar to PB0203.1_Zfp691_2/Jaspar(0.676)\n",
      "\t\t19 of 25 (1e-43) similar to PB0203.1_Zfp691_2/Jaspar(0.655)\n",
      "\t\t20 of 25 (1e-39) similar to PB0091.1_Zbtb3_1/Jaspar(0.837)\n",
      "\t\t21 of 25 (1e-26) similar to Oct4:Sox17(POU,Homeobox,HMG)/F9-Sox17-ChIP-Seq(GSE44553)/Homer(0.716)\n",
      "\t\t22 of 25 (1e-22) similar to POU3F4/MA0789.1/Jaspar(0.675)\n",
      "\t\t23 of 25 (1e-21) similar to Tcf7(HMG)/GM12878-TCF7-ChIP-Seq(Encode)/Homer(0.534)\n",
      "\t\t24 of 25 (1e-14) similar to RUNX2(Runt)/PCa-RUNX2-ChIP-Seq(GSE33889)/Homer(0.701)\n",
      "\t\t25 of 25 (1e-10) similar to GATA1/MA0035.4/Jaspar(0.620)\n",
      "\tJob finished - if results look good, please send beer to ..\n",
      "\n",
      "\tCleaning up tmp files...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "findMotifsGenome.pl peak/STAT1_6h_IFNa_summits.bed hg19 STAT1_6h_IFNa_homer -size 200 -preparsedDir ."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5fcb4081-ff63-44fd-aab1-2f9ff8d91d2f",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  },
  "toc-autonumbering": true
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
